{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Single-cell Bioinformatics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Over the course of your project, you'll have many ups and downs with your data, so we'll use the analogy of a romantic relationship to explain the different steps of single-cell bioinformatics analyses.\n",
    "\n",
    "## 1. Matchmaking: Getting the publicly available [deets](http://www.urbandictionary.com/define.php?term=deets) on your data\n",
    "\n",
    "\"Pubmed stalking\"... it's just like facebook stalking!\n",
    "\n",
    "* [1.0_Introduction_to_bioinformatics.ipynb](notebooks/1.0_Introduction_to_bioinformatics.ipynb)\n",
    "* [1.1_Overview_of_analysis_steps.ipynb](notebooks/1.1_Overview_of_analysis_steps.ipynb)\n",
    "* [1.2_Downloading_public_data_shalek2013.ipynb](notebooks/1.2_Downloading_public_data_Shalek2013.ipynb)\n",
    "* [1.3_Single-cell_overview_additional_reading.ipynb](notebooks/1.3_Single-cell_overview_additional_reading.ipynb)\n",
    "* [1.4_Unix.ipynb](notebooks/1.4_Unix.ipynb) - Optional additional exercises with the unix command line. If you're on Linux or Mac you can do this on your computer, if you have windows, do these on the Macs.\n",
    "\n",
    "### Homework\n",
    "- Spillover from what we didn't finish:\n",
    "    - Mapping/alignment spillover\n",
    "    - Downloading public data and filtering on expressed genes spillover\n",
    "- Find another single cell paper with GEO/ArrayExpress accession, download its data, and compare gene expression filtering strategies (Will use this dataset throughout the course)\n",
    "\n",
    "### Optional: Pandas from `.head()` to `.tail()`\n",
    "\n",
    "The package we'll be using to deal with matrices and dataframes in Python is called Pandas. Thoughout the course, I've tried to show some different applications of `pandas` but this is definitely not complete. For a full introduction, I recommend the following tutorial from Tom Augspurger.\n",
    "\n",
    "* [Video](https://www.youtube.com/watch?v=otCriSKVV_8)\n",
    "* [Notebooks](pandas_tutorial/)\n",
    "\n",
    "While this tutorial is aimed for newbies to Python and `pandas`, and thus the beginning would be review for intermediate to advanced Python and `pandas` users, the last few notebooks would be of interest to non-newbies.\n",
    "\n",
    "* [Groupby](pandas_tutorial/notebooks/4. Groupby.ipynb)\n",
    "    * Life-changing concept that has saved me hours of work. There's been many days where I've said to myself, ***\"I LOVE GROUPBY!!!!!!\"***\n",
    "* [Tidy Data](pandas_tutorial/notebooks/5. Tidy Data.ipynb)\n",
    "    * Another Awesome life-changing concept that helps you think about how to structure your data, even as you're making Excel files. Based off of [this](notebooks/papers/tidy-data.pdf) paper by Hadley Wickham, the author of many many dataframe manipulation packages in R.\n",
    "* [Pandas applied to Machine Learning and Statistics](pandas_tutorial/notebooks/6. For Stats & ML.ipynb)\n",
    "    * Categorical variables and transforming them to machine-learning friendly formats\n",
    "\n",
    "\n",
    "## 2. First date: Get your data's life story with dimensionality reduction\n",
    "\n",
    "* [2.0 Machine Learning Intro](sklearn_tutorial/notebooks/02.1-Machine-Learning-Intro.ipynb) [Jake Vanderplas' tutorial]\n",
    "* [2.1 Basic Principles in Machine Learning](sklearn_tutorial/notebooks/02.2-Basic-Principles.ipynb) [Jake Vanderplas' tutorial]\n",
    "* [2.2_Introduction_to_dimensionality_reduction.ipynb](notebooks/2.2_Introduction_to_dimensionality_reduction.ipynb)\n",
    "* [2.3_PCA](sklearn_tutorial/notebooks/04.1-Dimensionality-PCA.ipynb) [Jake Vanderplas' tutorial]\n",
    "* [2.4_ICA.ipynb](notebooks/2.4_ICA.ipynb)\n",
    "* [2.5_Manifold_learning.ipynb](notebooks/2.5_Manifold_learning.ipynb)\n",
    "* [2.6_Compare_dimensionality_reduction.ipynb](notebooks/2.6_Compare_dimensionality_reduction.ipynb)\n",
    "* [2.7_Apply_dimensionality_reduction_on_Shalek2013_Macaulay2016.ipynb](notebooks/2.7_Apply_dimensionality_reduction_on_Shalek2013_Macaulay2016.ipynb)\n",
    "* [2.8_Additional reading.ipynb](notebooks/2.8_Additional_reading.ipynb)\n",
    "* [2.9_tSNE_on_subsets_of_digits.ipynb](notebooks/2.9_tSNE_on_subsets_of_digits.ipynb)\n",
    "\n",
    "### Homework\n",
    "\n",
    "- Application spillover\n",
    "- Same single cell dataset, compare all dimensionality reduction algorithms\n",
    "\n",
    "## 3. One-month anniversary: Give your boo some clusters\n",
    "\n",
    "* [3.0_Introduction_to_clustering.ipynb](notebooks/3.0_Introduction_to_clustering.ipynb)\n",
    "* [3.1 $K$-means_clustering](sklearn_tutorial/notebooks/04.2-Clustering-KMeans.ipynb) [Jake Vanderplas' tutorial]\n",
    "* [3.2_Hierarchical_clustering.ipynb](notebooks/3.2_Hierarchical_clustering.ipynb)\n",
    "* [3.3_Apply_clustering_to_Shalek2013_Macaulay2016.ipynb](notebooks/3.3_Apply_clustering_to_Shalek2013_Macaulay2016.ipynb)\n",
    "* [3.4_Plotting_colors_and_evaluating_clustering.ipynb](notebooks/3.4_Plotting_colors_and_evaluating_clustering.ipynb)\n",
    "\n",
    "### Homework\n",
    "\n",
    "- Application spillover\n",
    "- Same dataset, compare cluster finding\n",
    "\n",
    "## 4. One-year anniversary: Find what makes your data tick using supervised learning\n",
    "\n",
    "* [4.0_Introduction_to_classifiers.ipynb](notebooks/4.0_Introduction_to_classifiers.ipynb)\n",
    "* [4.1_Overfitting.ipynb](notebooks/4.1_Overfitting.ipynb)\n",
    "* [4.1_Support_vector_machines](sklearn_tutorial/notebooks/03.1-Classification-SVMs.ipynb) [Jake Vanderplas' tutorial]\n",
    "* [4.2_Decision_trees](sklearn_tutorial/notebooks/03.2-Regression-Forests.ipynb) [Jake Vanderplas' tutorial]\n",
    "* [4.4_Apply_SVM_to_Shalek2013_clustered_heatmap.ipynb](notebooks/4.4_Apply_SVM_to_Shalek2013_clustered_heatmap.ipynb)\n",
    "* [4.5_Apply_SVM_to_Shalek2013_with_violinplots.ipynb](notebooks/4.5_Apply_SVM_to_Shalek2013_with_violinplots.ipynb)\n",
    "* [4.6_Assess_clustering_with_gene_ontology.ipynb](notebooks/4.6_Assess_clustering_with_gene_ontology.ipynb)\n",
    "* [4.7_Apply_tree_classifiers_to_Shalek2013_with_gene_ontology.ipynb](notebooks/4.7_Apply_tree_classifiers_to_Shalek2013_with_gene_ontology.ipynb)\n",
    "* [4.8_Apply_classifiers_to_Macaulay2016_with_gene_ontology.ipynb](notebooks/4.8_Apply_classifiers_to_Macaulay2016_with_gene_ontology.ipynb)\n",
    "\n",
    "### Homework\n",
    "\n",
    "- Application spillover\n",
    "- Same dataset, compare enriched genes in clusters\n",
    "\n",
    "## 5. Ten-year anniversary: Reflect on where you've been together with pseudotime ordering\n",
    "\n",
    "Pseudotime ordering is like biologically-driven \"regression\"\n",
    "\n",
    "* [5.0_Pseudotime_introduction.ipynb](notebooks/5.0_Pseudotime_introduction.ipynb)\n",
    "* [5.1_Pseudotime_ordering_algorithms_overiew.ipynb](notebooks/5.1_Pseudotime_ordering_algorithms_overiew.ipynb)\n",
    "\n",
    "\n",
    "## 6. Couples counseling: Dealing with technical noise and batch effects\n",
    "\n",
    "* [6.1_Dealing_with_technical_noise.ipynb](notebooks/6.1_Dealing_with_technical_noise.ipynb)\n",
    "* [6.2_Batch_Correction.ipynb](notebooks/6.2_Batch_Correction.ipynb)\n",
    "* [6.3_Technical_noise_additional_reading.ipynb](notebooks/6.3_Technical_noise_additional_reading.ipynb)\n",
    "\n",
    "## 7. 50-year anniversary: Advanced topics\n",
    "\n",
    "If you're already an experienced bioinformatician, you may be interested in working through the analyses steps of the papers assigned for the course. The simpler one is the Shalek2013 paper:\n",
    "\n",
    "* [7.2_Reproducing_Shalek2013_figures](notebooks/7.2_Reproducing_Shalek2013_figures.ipynb)\n",
    "\n",
    "\n",
    "More advanced is the Macaulay2016 paper, which includes pseudotime ordering and Bayesian modeling.\n",
    "\n",
    "* [7.0_Case_Study_Macaulay2016.ipynb](notebooks/7.0_Case_Study_Macaulay2016.ipynb)\n",
    "    * Links to the original notebooks supplied with the paper\n",
    "* [7.1_Playing_with_analysis_decisions_in_Macaulay2016.ipynb](notebooks/7.1_Playing_with_analysis_decisions_in_Macaulay2016.ipynb)\n",
    "    * Interactive widgets playing with PCA vs ICA vs MDS vs t-SNE, linkage methods, and distance metrics at key points of the Macaulay2016 analysis pipeline\n",
    "\n",
    "\n",
    "## 8. Plotting tips\n",
    "\n",
    "Tips for Python plotting with colors and such\n",
    "\n",
    "* [8.0_Plotting_tips.ipynb](8.0_Plotting_tips.ipynb)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (single-cell-bioinformatics)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
