{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/kirkreardon/Downloads/GSE67835_RAW\n"
     ]
    }
   ],
   "source": [
    "cd ~/Downloads/GSE67835_RAW/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GSM1657871_1772078217.C03.csv     GSM1658104_nochipID3.C69.csv.gz\r\n",
      "GSM1657871_1772078217.C03.csv.gz  GSM1658105_nochipID3.C71.csv.gz\r\n",
      "GSM1657872_1772078217.C04.csv.gz  GSM1658106_nochipID3.C72.csv.gz\r\n",
      "GSM1657873_1772078217.C06.csv.gz  GSM1658107_nochipID3.C74.csv.gz\r\n",
      "GSM1657874_1772078217.C07.csv.gz  GSM1658108_nochipID3.C75.csv.gz\r\n",
      "GSM1657875_1772078217.C08.csv.gz  GSM1658109_nochipID3.C76.csv.gz\r\n",
      "GSM1657876_1772078217.C09.csv.gz  GSM1658110_nochipID3.C80.csv.gz\r\n",
      "GSM1657877_1772078217.C14.csv.gz  GSM1658111_nochipID3.C81.csv.gz\r\n",
      "GSM1657878_1772078217.C16.csv.gz  GSM1658112_nochipID3.C84.csv.gz\r\n",
      "GSM1657879_1772078217.C17.csv.gz  GSM1658113_nochipID3.C85.csv.gz\r\n",
      "GSM1657880_1772078217.C18.csv.gz  GSM1658114_nochipID3.C86.csv.gz\r\n",
      "GSM1657881_1772078217.C20.csv.gz  GSM1658115_nochipID3.C91.csv.gz\r\n",
      "GSM1657882_1772078217.C23.csv.gz  GSM1658116_nochipID5.C04.csv.gz\r\n",
      "GSM1657883_1772078217.C28.csv.gz  GSM1658117_nochipID5.C05.csv.gz\r\n",
      "GSM1657884_1772078217.C29.csv.gz  GSM1658118_nochipID5.C18.csv.gz\r\n",
      "GSM1657885_1772078217.C32.csv.gz  GSM1658119_nochipID5.C19.csv.gz\r\n",
      "GSM1657886_1772078217.C33.csv.gz  GSM1658120_nochipID5.C42.csv.gz\r\n",
      "GSM1657887_1772078217.C39.csv.gz  GSM1658121_nochipID5.C44.csv.gz\r\n",
      "GSM1657888_1772078217.C40.csv.gz  GSM1658122_nochipID5.C45.csv.gz\r\n",
      "GSM1657889_1772078217.C47.csv.gz  GSM1658123_nochipID5.C54.csv.gz\r\n",
      "GSM1657890_1772078217.C52.csv.gz  GSM1658124_nochipID5.C65.csv.gz\r\n",
      "GSM1657891_1772078217.C56.csv.gz  GSM1658125_nochipID5.C72.csv.gz\r\n",
      "GSM1657892_1772078217.C58.csv.gz  GSM1658126_nochipID5.C91.csv.gz\r\n",
      "GSM1657893_1772078217.C59.csv.gz  GSM1658127_nochipID8.C01.csv.gz\r\n",
      "GSM1657894_1772078217.C60.csv.gz  GSM1658128_nochipID8.C02.csv.gz\r\n",
      "GSM1657895_1772078217.C61.csv.gz  GSM1658129_nochipID8.C05.csv.gz\r\n",
      "GSM1657896_1772078217.C66.csv.gz  GSM1658130_nochipID8.C06.csv.gz\r\n",
      "GSM1657897_1772078217.C72.csv.gz  GSM1658131_nochipID8.C08.csv.gz\r\n",
      "GSM1657898_1772078217.C80.csv.gz  GSM1658132_nochipID8.C11.csv.gz\r\n",
      "GSM1657899_1772078217.C87.csv.gz  GSM1658133_nochipID8.C12.csv.gz\r\n",
      "GSM1657900_1772078217.C89.csv.gz  GSM1658134_nochipID8.C13.csv.gz\r\n",
      "GSM1657901_1772078217.C91.csv.gz  GSM1658135_nochipID8.C14.csv.gz\r\n",
      "GSM1657902_1772078217.C94.csv.gz  GSM1658136_nochipID8.C17.csv.gz\r\n",
      "GSM1657903_1772078217.C96.csv.gz  GSM1658137_nochipID8.C19.csv.gz\r\n",
      "GSM1657904_1772078218.C05.csv.gz  GSM1658138_nochipID8.C20.csv.gz\r\n",
      "GSM1657905_1772078218.C08.csv.gz  GSM1658139_nochipID8.C22.csv.gz\r\n",
      "GSM1657906_1772078218.C09.csv.gz  GSM1658140_nochipID8.C23.csv.gz\r\n",
      "GSM1657907_1772078218.C13.csv.gz  GSM1658141_nochipID8.C25.csv.gz\r\n",
      "GSM1657908_1772078218.C18.csv.gz  GSM1658142_nochipID8.C28.csv.gz\r\n",
      "GSM1657909_1772078218.C23.csv.gz  GSM1658143_nochipID8.C29.csv.gz\r\n",
      "GSM1657910_1772078218.C27.csv.gz  GSM1658144_nochipID8.C30.csv.gz\r\n",
      "GSM1657911_1772078218.C29.csv.gz  GSM1658145_nochipID8.C32.csv.gz\r\n",
      "GSM1657912_1772078218.C32.csv.gz  GSM1658146_nochipID8.C36.csv.gz\r\n",
      "GSM1657913_1772078218.C34.csv.gz  GSM1658147_nochipID8.C37.csv.gz\r\n",
      "GSM1657914_1772078218.C43.csv.gz  GSM1658148_nochipID8.C39.csv.gz\r\n",
      "GSM1657915_1772078218.C44.csv.gz  GSM1658149_nochipID8.C40.csv.gz\r\n",
      "GSM1657916_1772078218.C47.csv.gz  GSM1658150_nochipID8.C41.csv.gz\r\n",
      "GSM1657917_1772078218.C48.csv.gz  GSM1658151_nochipID8.C42.csv.gz\r\n",
      "GSM1657918_1772078218.C53.csv.gz  GSM1658152_nochipID8.C43.csv.gz\r\n",
      "GSM1657919_1772078218.C55.csv.gz  GSM1658153_nochipID8.C45.csv.gz\r\n",
      "GSM1657920_1772078218.C56.csv.gz  GSM1658154_nochipID8.C48.csv.gz\r\n",
      "GSM1657921_1772078218.C58.csv.gz  GSM1658155_nochipID8.C49.csv.gz\r\n",
      "GSM1657922_1772078218.C61.csv.gz  GSM1658156_nochipID8.C50.csv.gz\r\n",
      "GSM1657923_1772078218.C71.csv.gz  GSM1658157_nochipID8.C52.csv.gz\r\n",
      "GSM1657924_1772078218.C85.csv.gz  GSM1658158_nochipID8.C53.csv.gz\r\n",
      "GSM1657925_1772078218.C88.csv.gz  GSM1658159_nochipID8.C54.csv.gz\r\n",
      "GSM1657926_1772078218.C89.csv.gz  GSM1658160_nochipID8.C55.csv.gz\r\n",
      "GSM1657927_1772078218.C93.csv.gz  GSM1658161_nochipID8.C56.csv.gz\r\n",
      "GSM1657928_1772078218.C95.csv.gz  GSM1658162_nochipID8.C57.csv.gz\r\n",
      "GSM1657929_1772078236.C11.csv.gz  GSM1658163_nochipID8.C60.csv.gz\r\n",
      "GSM1657930_1772078236.C12.csv.gz  GSM1658164_nochipID8.C61.csv.gz\r\n",
      "GSM1657931_1772078236.C16.csv.gz  GSM1658165_nochipID8.C64.csv.gz\r\n",
      "GSM1657932_1772078236.C17.csv.gz  GSM1658166_nochipID8.C65.csv.gz\r\n",
      "GSM1657933_1772078236.C19.csv.gz  GSM1658167_nochipID8.C66.csv.gz\r\n",
      "GSM1657934_1772078236.C25.csv.gz  GSM1658168_nochipID8.C67.csv.gz\r\n",
      "GSM1657935_1772078236.C26.csv.gz  GSM1658169_nochipID8.C70.csv.gz\r\n",
      "GSM1657936_1772078236.C27.csv.gz  GSM1658170_nochipID8.C75.csv.gz\r\n",
      "GSM1657937_1772078236.C28.csv.gz  GSM1658171_nochipID8.C77.csv.gz\r\n",
      "GSM1657938_1772078236.C30.csv.gz  GSM1658172_nochipID8.C78.csv.gz\r\n",
      "GSM1657939_1772078236.C31.csv.gz  GSM1658173_nochipID8.C79.csv.gz\r\n",
      "GSM1657940_1772078236.C32.csv.gz  GSM1658174_nochipID8.C81.csv.gz\r\n",
      "GSM1657941_1772078236.C33.csv.gz  GSM1658175_nochipID8.C82.csv.gz\r\n",
      "GSM1657942_1772078236.C34.csv.gz  GSM1658176_nochipID8.C84.csv.gz\r\n",
      "GSM1657943_1772078236.C36.csv.gz  GSM1658177_nochipID8.C85.csv.gz\r\n",
      "GSM1657944_1772078236.C38.csv.gz  GSM1658178_nochipID8.C87.csv.gz\r\n",
      "GSM1657945_1772078236.C40.csv.gz  GSM1658179_nochipID8.C88.csv.gz\r\n",
      "GSM1657946_1772078236.C41.csv.gz  GSM1658180_nochipID8.C90.csv.gz\r\n",
      "GSM1657947_1772078236.C45.csv.gz  GSM1658181_nochipID8.C92.csv.gz\r\n",
      "GSM1657948_1772078236.C46.csv.gz  GSM1658182_nochipID8.C95.csv.gz\r\n",
      "GSM1657949_1772078236.C47.csv.gz  GSM1658183_nochipID8.C96.csv.gz\r\n",
      "GSM1657950_1772078236.C49.csv.gz  GSM1658184_nochipID9.C01.csv.gz\r\n",
      "GSM1657951_1772078236.C54.csv.gz  GSM1658185_nochipID9.C02.csv.gz\r\n",
      "GSM1657952_1772078236.C56.csv.gz  GSM1658186_nochipID9.C04.csv.gz\r\n",
      "GSM1657953_1772078236.C57.csv.gz  GSM1658187_nochipID9.C09.csv.gz\r\n",
      "GSM1657954_1772078236.C61.csv.gz  GSM1658188_nochipID9.C14.csv.gz\r\n",
      "GSM1657955_1772078236.C65.csv.gz  GSM1658189_nochipID9.C18.csv.gz\r\n",
      "GSM1657956_1772078236.C66.csv.gz  GSM1658190_nochipID9.C26.csv.gz\r\n",
      "GSM1657957_1772078236.C70.csv.gz  GSM1658191_nochipID9.C37.csv.gz\r\n",
      "GSM1657958_1772078236.C74.csv.gz  GSM1658192_nochipID9.C42.csv.gz\r\n",
      "GSM1657959_1772078236.C77.csv.gz  GSM1658193_nochipID9.C57.csv.gz\r\n",
      "GSM1657960_1772078236.C89.csv.gz  GSM1658194_nochipID9.C59.csv.gz\r\n",
      "GSM1657961_1772078236.C90.csv.gz  GSM1658195_nochipID9.C62.csv.gz\r\n",
      "GSM1657962_1772078236.C93.csv.gz  GSM1658196_nochipID9.C65.csv.gz\r\n",
      "GSM1657963_1772078237.C06.csv.gz  GSM1658197_nochipID9.C74.csv.gz\r\n",
      "GSM1657964_1772078237.C08.csv.gz  GSM1658198_nochipID9.C75.csv.gz\r\n",
      "GSM1657965_1772078237.C11.csv.gz  GSM1658199_nochipID9.C84.csv.gz\r\n",
      "GSM1657966_1772078237.C13.csv.gz  GSM1658200_nochipID9.C85.csv.gz\r\n",
      "GSM1657967_1772078237.C14.csv.gz  GSM1658201_nochipID9.C87.csv.gz\r\n",
      "GSM1657968_1772078237.C15.csv.gz  GSM1658202_nochipID9.C92.csv.gz\r\n",
      "GSM1657969_1772078237.C19.csv.gz  GSM1658203_nochipID10.C02.csv.gz\r\n",
      "GSM1657970_1772078237.C21.csv.gz  GSM1658204_nochipID10.C03.csv.gz\r\n",
      "GSM1657971_1772078237.C26.csv.gz  GSM1658205_nochipID10.C05.csv.gz\r\n",
      "GSM1657972_1772078237.C28.csv.gz  GSM1658206_nochipID10.C07.csv.gz\r\n",
      "GSM1657973_1772078237.C29.csv.gz  GSM1658207_nochipID10.C11.csv.gz\r\n",
      "GSM1657974_1772078237.C31.csv.gz  GSM1658208_nochipID10.C16.csv.gz\r\n",
      "GSM1657975_1772078237.C40.csv.gz  GSM1658209_nochipID10.C17.csv.gz\r\n",
      "GSM1657976_1772078237.C42.csv.gz  GSM1658210_nochipID10.C19.csv.gz\r\n",
      "GSM1657977_1772078237.C44.csv.gz  GSM1658211_nochipID10.C25.csv.gz\r\n",
      "GSM1657978_1772078237.C47.csv.gz  GSM1658212_nochipID10.C26.csv.gz\r\n",
      "GSM1657979_1772078237.C49.csv.gz  GSM1658213_nochipID10.C37.csv.gz\r\n",
      "GSM1657980_1772078237.C52.csv.gz  GSM1658214_nochipID10.C39.csv.gz\r\n",
      "GSM1657981_1772078237.C55.csv.gz  GSM1658215_nochipID10.C42.csv.gz\r\n",
      "GSM1657982_1772078237.C56.csv.gz  GSM1658216_nochipID10.C44.csv.gz\r\n",
      "GSM1657983_1772078237.C58.csv.gz  GSM1658217_nochipID10.C47.csv.gz\r\n",
      "GSM1657984_1772078237.C60.csv.gz  GSM1658218_nochipID10.C49.csv.gz\r\n",
      "GSM1657985_1772078237.C61.csv.gz  GSM1658219_nochipID10.C50.csv.gz\r\n",
      "GSM1657986_1772078237.C62.csv.gz  GSM1658220_nochipID10.C53.csv.gz\r\n",
      "GSM1657987_1772078237.C74.csv.gz  GSM1658221_nochipID10.C58.csv.gz\r\n",
      "GSM1657988_1772078237.C79.csv.gz  GSM1658222_nochipID10.C65.csv.gz\r\n",
      "GSM1657989_1772078237.C83.csv.gz  GSM1658223_nochipID10.C74.csv.gz\r\n",
      "GSM1657990_1772078237.C84.csv.gz  GSM1658224_nochipID10.C75.csv.gz\r\n",
      "GSM1657991_1772078237.C92.csv.gz  GSM1658225_nochipID10.C85.csv.gz\r\n",
      "GSM1657992_nochipID12.C05.csv.gz  GSM1658226_nochipID10.C88.csv.gz\r\n",
      "GSM1657993_nochipID12.C23.csv.gz  GSM1658227_nochipID10.C93.csv.gz\r\n",
      "GSM1657994_nochipID12.C73.csv.gz  GSM1658228_nochipID10.C94.csv.gz\r\n",
      "GSM1657995_nochipID12.C84.csv.gz  GSM1658229_nochipID11.C02.csv.gz\r\n",
      "GSM1657996_nochipID12.C87.csv.gz  GSM1658230_nochipID11.C03.csv.gz\r\n",
      "GSM1657997_nochipID14.C08.csv.gz  GSM1658231_nochipID11.C06.csv.gz\r\n",
      "GSM1657998_nochipID14.C13.csv.gz  GSM1658232_nochipID11.C07.csv.gz\r\n",
      "GSM1657999_nochipID14.C29.csv.gz  GSM1658233_nochipID11.C08.csv.gz\r\n",
      "GSM1658000_nochipID14.C45.csv.gz  GSM1658234_nochipID11.C10.csv.gz\r\n",
      "GSM1658001_nochipID14.C89.csv.gz  GSM1658235_nochipID11.C15.csv.gz\r\n",
      "GSM1658002_nochipID15.C20.csv.gz  GSM1658236_nochipID11.C17.csv.gz\r\n",
      "GSM1658003_nochipID15.C54.csv.gz  GSM1658237_nochipID11.C19.csv.gz\r\n",
      "GSM1658004_nochipID15.C69.csv.gz  GSM1658238_nochipID11.C20.csv.gz\r\n",
      "GSM1658005_nochipID15.C86.csv.gz  GSM1658239_nochipID11.C21.csv.gz\r\n",
      "GSM1658006_nochipID2.C01.csv.gz   GSM1658240_nochipID11.C22.csv.gz\r\n",
      "GSM1658007_nochipID2.C02.csv.gz   GSM1658241_nochipID11.C23.csv.gz\r\n",
      "GSM1658008_nochipID2.C03.csv.gz   GSM1658242_nochipID11.C25.csv.gz\r\n",
      "GSM1658009_nochipID2.C07.csv.gz   GSM1658243_nochipID11.C30.csv.gz\r\n",
      "GSM1658010_nochipID2.C08.csv.gz   GSM1658244_nochipID11.C37.csv.gz\r\n",
      "GSM1658011_nochipID2.C09.csv.gz   GSM1658245_nochipID11.C48.csv.gz\r\n",
      "GSM1658012_nochipID2.C10.csv.gz   GSM1658246_nochipID11.C49.csv.gz\r\n",
      "GSM1658013_nochipID2.C11.csv.gz   GSM1658247_nochipID11.C50.csv.gz\r\n",
      "GSM1658014_nochipID2.C12.csv.gz   GSM1658248_nochipID11.C53.csv.gz\r\n",
      "GSM1658015_nochipID2.C13.csv.gz   GSM1658249_nochipID11.C54.csv.gz\r\n",
      "GSM1658016_nochipID2.C14.csv.gz   GSM1658251_nochipID11.C55.csv.gz\r\n",
      "GSM1658017_nochipID2.C15.csv.gz   GSM1658253_nochipID11.C56.csv.gz\r\n",
      "GSM1658018_nochipID2.C16.csv.gz   GSM1658255_nochipID11.C57.csv.gz\r\n",
      "GSM1658019_nochipID2.C17.csv.gz   GSM1658257_nochipID11.C58.csv.gz\r\n",
      "GSM1658020_nochipID2.C18.csv.gz   GSM1658259_nochipID11.C59.csv.gz\r\n",
      "GSM1658021_nochipID2.C19.csv.gz   GSM1658262_nochipID11.C60.csv.gz\r\n",
      "GSM1658022_nochipID2.C21.csv.gz   GSM1658264_nochipID11.C61.csv.gz\r\n",
      "GSM1658023_nochipID2.C22.csv.gz   GSM1658266_nochipID11.C62.csv.gz\r\n",
      "GSM1658024_nochipID2.C24.csv.gz   GSM1658268_nochipID11.C63.csv.gz\r\n",
      "GSM1658025_nochipID2.C26.csv.gz   GSM1658270_nochipID11.C64.csv.gz\r\n",
      "GSM1658026_nochipID2.C27.csv.gz   GSM1658272_nochipID11.C66.csv.gz\r\n",
      "GSM1658027_nochipID2.C28.csv.gz   GSM1658275_nochipID11.C68.csv.gz\r\n",
      "GSM1658028_nochipID2.C29.csv.gz   GSM1658277_nochipID11.C69.csv.gz\r\n",
      "GSM1658029_nochipID2.C30.csv.gz   GSM1658279_nochipID11.C75.csv.gz\r\n",
      "GSM1658030_nochipID2.C31.csv.gz   GSM1658281_nochipID11.C76.csv.gz\r\n",
      "GSM1658031_nochipID2.C32.csv.gz   GSM1658284_nochipID11.C77.csv.gz\r\n",
      "GSM1658032_nochipID2.C33.csv.gz   GSM1658286_nochipID11.C78.csv.gz\r\n",
      "GSM1658033_nochipID2.C34.csv.gz   GSM1658288_nochipID11.C79.csv.gz\r\n",
      "GSM1658034_nochipID2.C36.csv.gz   GSM1658290_nochipID11.C81.csv.gz\r\n",
      "GSM1658035_nochipID2.C38.csv.gz   GSM1658292_nochipID11.C82.csv.gz\r\n",
      "GSM1658036_nochipID2.C39.csv.gz   GSM1658294_nochipID11.C83.csv.gz\r\n",
      "GSM1658037_nochipID2.C40.csv.gz   GSM1658297_nochipID11.C84.csv.gz\r\n",
      "GSM1658038_nochipID2.C41.csv.gz   GSM1658299_nochipID11.C86.csv.gz\r\n",
      "GSM1658039_nochipID2.C42.csv.gz   GSM1658301_nochipID11.C91.csv.gz\r\n",
      "GSM1658040_nochipID2.C43.csv.gz   GSM1658304_nochipID11.C93.csv.gz\r\n",
      "GSM1658041_nochipID2.C44.csv.gz   GSM1658305_nochipID13.C07.csv.gz\r\n",
      "GSM1658042_nochipID2.C45.csv.gz   GSM1658306_nochipID13.C11.csv.gz\r\n",
      "GSM1658043_nochipID2.C46.csv.gz   GSM1658307_nochipID13.C12.csv.gz\r\n",
      "GSM1658044_nochipID2.C47.csv.gz   GSM1658308_nochipID13.C14.csv.gz\r\n",
      "GSM1658045_nochipID2.C48.csv.gz   GSM1658309_nochipID13.C15.csv.gz\r\n",
      "GSM1658046_nochipID2.C49.csv.gz   GSM1658310_nochipID13.C17.csv.gz\r\n",
      "GSM1658047_nochipID2.C50.csv.gz   GSM1658311_nochipID13.C21.csv.gz\r\n",
      "GSM1658048_nochipID2.C51.csv.gz   GSM1658312_nochipID13.C25.csv.gz\r\n",
      "GSM1658049_nochipID2.C52.csv.gz   GSM1658313_nochipID13.C28.csv.gz\r\n",
      "GSM1658050_nochipID2.C56.csv.gz   GSM1658314_nochipID13.C32.csv.gz\r\n",
      "GSM1658051_nochipID2.C57.csv.gz   GSM1658315_nochipID13.C36.csv.gz\r\n",
      "GSM1658052_nochipID2.C58.csv.gz   GSM1658316_nochipID13.C45.csv.gz\r\n",
      "GSM1658053_nochipID2.C59.csv.gz   GSM1658317_nochipID13.C46.csv.gz\r\n",
      "GSM1658054_nochipID2.C60.csv.gz   GSM1658318_nochipID13.C48.csv.gz\r\n",
      "GSM1658055_nochipID2.C61.csv.gz   GSM1658319_nochipID13.C50.csv.gz\r\n",
      "GSM1658056_nochipID2.C62.csv.gz   GSM1658320_nochipID13.C55.csv.gz\r\n",
      "GSM1658057_nochipID2.C63.csv.gz   GSM1658321_nochipID13.C57.csv.gz\r\n",
      "GSM1658058_nochipID2.C64.csv.gz   GSM1658322_nochipID13.C58.csv.gz\r\n",
      "GSM1658059_nochipID2.C65.csv.gz   GSM1658323_nochipID13.C59.csv.gz\r\n",
      "GSM1658060_nochipID2.C66.csv.gz   GSM1658324_nochipID13.C61.csv.gz\r\n",
      "GSM1658061_nochipID2.C67.csv.gz   GSM1658325_nochipID13.C63.csv.gz\r\n",
      "GSM1658062_nochipID2.C68.csv.gz   GSM1658326_nochipID13.C64.csv.gz\r\n",
      "GSM1658063_nochipID2.C69.csv.gz   GSM1658327_nochipID13.C66.csv.gz\r\n",
      "GSM1658064_nochipID2.C71.csv.gz   GSM1658328_nochipID13.C67.csv.gz\r\n",
      "GSM1658065_nochipID2.C73.csv.gz   GSM1658329_nochipID13.C68.csv.gz\r\n",
      "GSM1658066_nochipID2.C75.csv.gz   GSM1658330_nochipID13.C72.csv.gz\r\n",
      "GSM1658067_nochipID2.C76.csv.gz   GSM1658331_nochipID13.C73.csv.gz\r\n",
      "GSM1658068_nochipID2.C77.csv.gz   GSM1658332_nochipID13.C74.csv.gz\r\n",
      "GSM1658069_nochipID2.C78.csv.gz   GSM1658333_nochipID13.C75.csv.gz\r\n",
      "GSM1658070_nochipID2.C79.csv.gz   GSM1658334_nochipID13.C80.csv.gz\r\n",
      "GSM1658071_nochipID2.C80.csv.gz   GSM1658335_nochipID13.C84.csv.gz\r\n",
      "GSM1658072_nochipID2.C81.csv.gz   GSM1658336_nochipID13.C87.csv.gz\r\n",
      "GSM1658073_nochipID2.C82.csv.gz   GSM1658337_nochipID13.C93.csv.gz\r\n",
      "GSM1658074_nochipID2.C83.csv.gz   GSM1658338_nochipID4.C03.csv.gz\r\n",
      "GSM1658075_nochipID2.C84.csv.gz   GSM1658339_nochipID4.C04.csv.gz\r\n",
      "GSM1658076_nochipID2.C85.csv.gz   GSM1658340_nochipID4.C10.csv.gz\r\n",
      "GSM1658077_nochipID2.C86.csv.gz   GSM1658341_nochipID4.C14.csv.gz\r\n",
      "GSM1658078_nochipID2.C87.csv.gz   GSM1658342_nochipID4.C18.csv.gz\r\n",
      "GSM1658079_nochipID2.C89.csv.gz   GSM1658343_nochipID4.C19.csv.gz\r\n",
      "GSM1658080_nochipID2.C90.csv.gz   GSM1658344_nochipID4.C20.csv.gz\r\n",
      "GSM1658081_nochipID2.C92.csv.gz   GSM1658345_nochipID4.C21.csv.gz\r\n",
      "GSM1658082_nochipID2.C93.csv.gz   GSM1658346_nochipID4.C33.csv.gz\r\n",
      "GSM1658083_nochipID3.C01.csv.gz   GSM1658347_nochipID4.C34.csv.gz\r\n",
      "GSM1658084_nochipID3.C05.csv.gz   GSM1658348_nochipID4.C38.csv.gz\r\n",
      "GSM1658085_nochipID3.C06.csv.gz   GSM1658349_nochipID4.C39.csv.gz\r\n",
      "GSM1658086_nochipID3.C07.csv.gz   GSM1658350_nochipID4.C41.csv.gz\r\n",
      "GSM1658087_nochipID3.C09.csv.gz   GSM1658351_nochipID4.C42.csv.gz\r\n",
      "GSM1658088_nochipID3.C11.csv.gz   GSM1658352_nochipID4.C44.csv.gz\r\n",
      "GSM1658089_nochipID3.C12.csv.gz   GSM1658353_nochipID4.C49.csv.gz\r\n",
      "GSM1658090_nochipID3.C14.csv.gz   GSM1658354_nochipID4.C52.csv.gz\r\n",
      "GSM1658091_nochipID3.C23.csv.gz   GSM1658355_nochipID4.C53.csv.gz\r\n",
      "GSM1658092_nochipID3.C25.csv.gz   GSM1658356_nochipID4.C59.csv.gz\r\n",
      "GSM1658093_nochipID3.C32.csv.gz   GSM1658357_nochipID4.C62.csv.gz\r\n",
      "GSM1658094_nochipID3.C37.csv.gz   GSM1658358_nochipID4.C63.csv.gz\r\n",
      "GSM1658095_nochipID3.C38.csv.gz   GSM1658359_nochipID4.C66.csv.gz\r\n",
      "GSM1658096_nochipID3.C51.csv.gz   GSM1658360_nochipID4.C74.csv.gz\r\n",
      "GSM1658097_nochipID3.C52.csv.gz   GSM1658361_nochipID4.C77.csv.gz\r\n",
      "GSM1658098_nochipID3.C53.csv.gz   GSM1658362_nochipID4.C78.csv.gz\r\n",
      "GSM1658099_nochipID3.C54.csv.gz   GSM1658363_nochipID4.C84.csv.gz\r\n",
      "GSM1658100_nochipID3.C56.csv.gz   GSM1658364_nochipID4.C89.csv.gz\r\n",
      "GSM1658101_nochipID3.C61.csv.gz   GSM1658365_nochipID4.C95.csv.gz\r\n",
      "GSM1658102_nochipID3.C62.csv.gz   GSM1658366_nochipID4.C96.csv.gz\r\n",
      "GSM1658103_nochipID3.C65.csv.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So now we have the unzipped version of the file, \"`GSE41265_allGenesTPM.txt`\". I wonder how much space they saved by zipping it?\n",
    "\n",
    "Let's use the flags \"`-l`\" for \"long listing\" which will show us the sizes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.0_Introduction_to_bioinformatics.ipynb\r\n",
      "1.1_Overview_of_analysis_steps.ipynb\r\n",
      "1.2_Downloading_public_data_Shalek2013.ipynb\r\n",
      "1.3_Single-cell_overview_additional_reading.ipynb\r\n",
      "1.X_unused_code.ipynb\r\n",
      "2.2_Introduction_to_dimensionality_reduction.ipynb\r\n",
      "2.4_ICA.ipynb\r\n",
      "2.5_Manifold_learning.ipynb\r\n",
      "2.6_Compare_dimensionality_reduction.ipynb\r\n",
      "2.7_Apply_dimensionality_reduction_on_Shalek2013_Macaulay2016.ipynb\r\n",
      "2.8_Additional_reading.ipynb\r\n",
      "2016-06-09_darmanis2015_concatenate_data.ipynb\r\n",
      "3.0_Introduction_to_clustering.ipynb\r\n",
      "3.1_Classifiers.ipynb\r\n",
      "3.2_Hierarchical_clustering.ipynb\r\n",
      "3.3_Apply_clustering_to_Shalek2013_Macaulay2016.ipynb\r\n",
      "3.X_Unused.ipynb\r\n",
      "4.0_Introduction_to_classifiers.ipynb\r\n",
      "4.3_Apply_classifiers_to_Shalek2013_Macaulay2016_with_gene_ontology.ipynb\r\n",
      "4.4_Additional_reading.ipynb\r\n",
      "5.0_Pseudotime_introduction.ipynb\r\n",
      "5.1_Pseudotime_ordering_algorithms_overiew.ipynb\r\n",
      "5.3_Pseudotime_additional_reading.ipynb\r\n",
      "6.1_Dealing_with_technical_noise.ipynb\r\n",
      "6.2_Batch_Correction.ipynb\r\n",
      "6.3_Technical_noise_additional_reading.ipynb\r\n",
      "7.0_Case_Study_Macaulay2016.ipynb\r\n",
      "7.1_Playing_with_analysis_decisions_in_Macaulay2016.ipynb\r\n",
      "7.2_Reproducing_Shalek2013_figures.ipynb\r\n",
      "8.0_Plotting_tips.ipynb\r\n",
      "GSE41265_allGenesTPM.txt\r\n",
      "GSE41265_allGenesTPM.txt.gz\r\n",
      "\u001b[1m\u001b[36mcombat_py\u001b[m\u001b[m/\r\n",
      "\u001b[1m\u001b[36mfigures\u001b[m\u001b[m/\r\n",
      "\u001b[1m\u001b[36mpapers\u001b[m\u001b[m/\r\n",
      "shalek2013_pca.pdf\r\n"
     ]
    }
   ],
   "source": [
    "ls -1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "oof, this is in pure bytes and I can't convert to multiples of 1024 easily in my head (1024 bytes = 1 kilobyte, 1024 kilobytes = 1 megabtye, etc -  the 1000/byte is a lie that the hard drive companies use!). So let's use the `-h` flag, which tells the computer to do th conversion for us. We can combine multiple flags with the same dash, so\n",
    "\n",
    "    ls -l -h\n",
    "\n",
    "Can be shortened to:\n",
    "\n",
    "    ls -lh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     466\r\n"
     ]
    }
   ],
   "source": [
    "! ls -1 | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Apple gzip 251\r\n",
      "usage: gunzip [-123456789acdfhklLNnqrtVv] [-S .suffix] [<file> [<file> ...]]\r\n",
      " -1 --fast            fastest (worst) compression\r\n",
      " -2 .. -8             set compression level\r\n",
      " -9 --best            best (slowest) compression\r\n",
      " -c --stdout          write to stdout, keep original files\r\n",
      "    --to-stdout\r\n",
      " -d --decompress      uncompress files\r\n",
      "    --uncompress\r\n",
      " -f --force           force overwriting & compress links\r\n",
      " -h --help            display this help\r\n",
      " -k --keep            don't delete input files during operation\r\n",
      " -l --list            list compressed file contents\r\n",
      " -N --name            save or restore original file name and time stamp\r\n",
      " -n --no-name         don't save original file name or time stamp\r\n",
      " -q --quiet           output no warnings\r\n",
      " -r --recursive       recursively compress files in directories\r\n",
      " -S .suf              use suffix .suf instead of .gz\r\n",
      "    --suffix .suf\r\n",
      " -t --test            test compressed file\r\n",
      " -V --version         display program version\r\n",
      " -v --verbose         print extra statistics\r\n"
     ]
    }
   ],
   "source": [
    "! gunzip --help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "See, \"`GSE41265_allGenesTPM.txt.gz`\" is there!\n",
    "\n",
    "Since the file ends in \"`.gz`\", this tells us its a \"gnu-zipped\" or \"gzipped\" (\"gee-zipped\") file, which is a specific flavor of \"zipping\" or compressing a file. We need to use a gnu-zipping-aware program to decompress the file, which is \"`gunzip`\" (\"gnu-unzip\").\n",
    "\n",
    "Run the next cell to unzip the file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "! gunzip -f *gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "3+3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'beyonce'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asdf = 'beyonce'\n",
    "asdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'beyonce runs the world'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asdf + ' runs the world'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's \"`ls`\" again to see what files have changed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GSM1657871_1772078217.C03.csv  GSM1658104_nochipID3.C69.csv\r\n",
      "GSM1657872_1772078217.C04.csv  GSM1658105_nochipID3.C71.csv\r\n",
      "GSM1657873_1772078217.C06.csv  GSM1658106_nochipID3.C72.csv\r\n",
      "GSM1657874_1772078217.C07.csv  GSM1658107_nochipID3.C74.csv\r\n",
      "GSM1657875_1772078217.C08.csv  GSM1658108_nochipID3.C75.csv\r\n",
      "GSM1657876_1772078217.C09.csv  GSM1658109_nochipID3.C76.csv\r\n",
      "GSM1657877_1772078217.C14.csv  GSM1658110_nochipID3.C80.csv\r\n",
      "GSM1657878_1772078217.C16.csv  GSM1658111_nochipID3.C81.csv\r\n",
      "GSM1657879_1772078217.C17.csv  GSM1658112_nochipID3.C84.csv\r\n",
      "GSM1657880_1772078217.C18.csv  GSM1658113_nochipID3.C85.csv\r\n",
      "GSM1657881_1772078217.C20.csv  GSM1658114_nochipID3.C86.csv\r\n",
      "GSM1657882_1772078217.C23.csv  GSM1658115_nochipID3.C91.csv\r\n",
      "GSM1657883_1772078217.C28.csv  GSM1658116_nochipID5.C04.csv\r\n",
      "GSM1657884_1772078217.C29.csv  GSM1658117_nochipID5.C05.csv\r\n",
      "GSM1657885_1772078217.C32.csv  GSM1658118_nochipID5.C18.csv\r\n",
      "GSM1657886_1772078217.C33.csv  GSM1658119_nochipID5.C19.csv\r\n",
      "GSM1657887_1772078217.C39.csv  GSM1658120_nochipID5.C42.csv\r\n",
      "GSM1657888_1772078217.C40.csv  GSM1658121_nochipID5.C44.csv\r\n",
      "GSM1657889_1772078217.C47.csv  GSM1658122_nochipID5.C45.csv\r\n",
      "GSM1657890_1772078217.C52.csv  GSM1658123_nochipID5.C54.csv\r\n",
      "GSM1657891_1772078217.C56.csv  GSM1658124_nochipID5.C65.csv\r\n",
      "GSM1657892_1772078217.C58.csv  GSM1658125_nochipID5.C72.csv\r\n",
      "GSM1657893_1772078217.C59.csv  GSM1658126_nochipID5.C91.csv\r\n",
      "GSM1657894_1772078217.C60.csv  GSM1658127_nochipID8.C01.csv\r\n",
      "GSM1657895_1772078217.C61.csv  GSM1658128_nochipID8.C02.csv\r\n",
      "GSM1657896_1772078217.C66.csv  GSM1658129_nochipID8.C05.csv\r\n",
      "GSM1657897_1772078217.C72.csv  GSM1658130_nochipID8.C06.csv\r\n",
      "GSM1657898_1772078217.C80.csv  GSM1658131_nochipID8.C08.csv\r\n",
      "GSM1657899_1772078217.C87.csv  GSM1658132_nochipID8.C11.csv\r\n",
      "GSM1657900_1772078217.C89.csv  GSM1658133_nochipID8.C12.csv\r\n",
      "GSM1657901_1772078217.C91.csv  GSM1658134_nochipID8.C13.csv\r\n",
      "GSM1657902_1772078217.C94.csv  GSM1658135_nochipID8.C14.csv\r\n",
      "GSM1657903_1772078217.C96.csv  GSM1658136_nochipID8.C17.csv\r\n",
      "GSM1657904_1772078218.C05.csv  GSM1658137_nochipID8.C19.csv\r\n",
      "GSM1657905_1772078218.C08.csv  GSM1658138_nochipID8.C20.csv\r\n",
      "GSM1657906_1772078218.C09.csv  GSM1658139_nochipID8.C22.csv\r\n",
      "GSM1657907_1772078218.C13.csv  GSM1658140_nochipID8.C23.csv\r\n",
      "GSM1657908_1772078218.C18.csv  GSM1658141_nochipID8.C25.csv\r\n",
      "GSM1657909_1772078218.C23.csv  GSM1658142_nochipID8.C28.csv\r\n",
      "GSM1657910_1772078218.C27.csv  GSM1658143_nochipID8.C29.csv\r\n",
      "GSM1657911_1772078218.C29.csv  GSM1658144_nochipID8.C30.csv\r\n",
      "GSM1657912_1772078218.C32.csv  GSM1658145_nochipID8.C32.csv\r\n",
      "GSM1657913_1772078218.C34.csv  GSM1658146_nochipID8.C36.csv\r\n",
      "GSM1657914_1772078218.C43.csv  GSM1658147_nochipID8.C37.csv\r\n",
      "GSM1657915_1772078218.C44.csv  GSM1658148_nochipID8.C39.csv\r\n",
      "GSM1657916_1772078218.C47.csv  GSM1658149_nochipID8.C40.csv\r\n",
      "GSM1657917_1772078218.C48.csv  GSM1658150_nochipID8.C41.csv\r\n",
      "GSM1657918_1772078218.C53.csv  GSM1658151_nochipID8.C42.csv\r\n",
      "GSM1657919_1772078218.C55.csv  GSM1658152_nochipID8.C43.csv\r\n",
      "GSM1657920_1772078218.C56.csv  GSM1658153_nochipID8.C45.csv\r\n",
      "GSM1657921_1772078218.C58.csv  GSM1658154_nochipID8.C48.csv\r\n",
      "GSM1657922_1772078218.C61.csv  GSM1658155_nochipID8.C49.csv\r\n",
      "GSM1657923_1772078218.C71.csv  GSM1658156_nochipID8.C50.csv\r\n",
      "GSM1657924_1772078218.C85.csv  GSM1658157_nochipID8.C52.csv\r\n",
      "GSM1657925_1772078218.C88.csv  GSM1658158_nochipID8.C53.csv\r\n",
      "GSM1657926_1772078218.C89.csv  GSM1658159_nochipID8.C54.csv\r\n",
      "GSM1657927_1772078218.C93.csv  GSM1658160_nochipID8.C55.csv\r\n",
      "GSM1657928_1772078218.C95.csv  GSM1658161_nochipID8.C56.csv\r\n",
      "GSM1657929_1772078236.C11.csv  GSM1658162_nochipID8.C57.csv\r\n",
      "GSM1657930_1772078236.C12.csv  GSM1658163_nochipID8.C60.csv\r\n",
      "GSM1657931_1772078236.C16.csv  GSM1658164_nochipID8.C61.csv\r\n",
      "GSM1657932_1772078236.C17.csv  GSM1658165_nochipID8.C64.csv\r\n",
      "GSM1657933_1772078236.C19.csv  GSM1658166_nochipID8.C65.csv\r\n",
      "GSM1657934_1772078236.C25.csv  GSM1658167_nochipID8.C66.csv\r\n",
      "GSM1657935_1772078236.C26.csv  GSM1658168_nochipID8.C67.csv\r\n",
      "GSM1657936_1772078236.C27.csv  GSM1658169_nochipID8.C70.csv\r\n",
      "GSM1657937_1772078236.C28.csv  GSM1658170_nochipID8.C75.csv\r\n",
      "GSM1657938_1772078236.C30.csv  GSM1658171_nochipID8.C77.csv\r\n",
      "GSM1657939_1772078236.C31.csv  GSM1658172_nochipID8.C78.csv\r\n",
      "GSM1657940_1772078236.C32.csv  GSM1658173_nochipID8.C79.csv\r\n",
      "GSM1657941_1772078236.C33.csv  GSM1658174_nochipID8.C81.csv\r\n",
      "GSM1657942_1772078236.C34.csv  GSM1658175_nochipID8.C82.csv\r\n",
      "GSM1657943_1772078236.C36.csv  GSM1658176_nochipID8.C84.csv\r\n",
      "GSM1657944_1772078236.C38.csv  GSM1658177_nochipID8.C85.csv\r\n",
      "GSM1657945_1772078236.C40.csv  GSM1658178_nochipID8.C87.csv\r\n",
      "GSM1657946_1772078236.C41.csv  GSM1658179_nochipID8.C88.csv\r\n",
      "GSM1657947_1772078236.C45.csv  GSM1658180_nochipID8.C90.csv\r\n",
      "GSM1657948_1772078236.C46.csv  GSM1658181_nochipID8.C92.csv\r\n",
      "GSM1657949_1772078236.C47.csv  GSM1658182_nochipID8.C95.csv\r\n",
      "GSM1657950_1772078236.C49.csv  GSM1658183_nochipID8.C96.csv\r\n",
      "GSM1657951_1772078236.C54.csv  GSM1658184_nochipID9.C01.csv\r\n",
      "GSM1657952_1772078236.C56.csv  GSM1658185_nochipID9.C02.csv\r\n",
      "GSM1657953_1772078236.C57.csv  GSM1658186_nochipID9.C04.csv\r\n",
      "GSM1657954_1772078236.C61.csv  GSM1658187_nochipID9.C09.csv\r\n",
      "GSM1657955_1772078236.C65.csv  GSM1658188_nochipID9.C14.csv\r\n",
      "GSM1657956_1772078236.C66.csv  GSM1658189_nochipID9.C18.csv\r\n",
      "GSM1657957_1772078236.C70.csv  GSM1658190_nochipID9.C26.csv\r\n",
      "GSM1657958_1772078236.C74.csv  GSM1658191_nochipID9.C37.csv\r\n",
      "GSM1657959_1772078236.C77.csv  GSM1658192_nochipID9.C42.csv\r\n",
      "GSM1657960_1772078236.C89.csv  GSM1658193_nochipID9.C57.csv\r\n",
      "GSM1657961_1772078236.C90.csv  GSM1658194_nochipID9.C59.csv\r\n",
      "GSM1657962_1772078236.C93.csv  GSM1658195_nochipID9.C62.csv\r\n",
      "GSM1657963_1772078237.C06.csv  GSM1658196_nochipID9.C65.csv\r\n",
      "GSM1657964_1772078237.C08.csv  GSM1658197_nochipID9.C74.csv\r\n",
      "GSM1657965_1772078237.C11.csv  GSM1658198_nochipID9.C75.csv\r\n",
      "GSM1657966_1772078237.C13.csv  GSM1658199_nochipID9.C84.csv\r\n",
      "GSM1657967_1772078237.C14.csv  GSM1658200_nochipID9.C85.csv\r\n",
      "GSM1657968_1772078237.C15.csv  GSM1658201_nochipID9.C87.csv\r\n",
      "GSM1657969_1772078237.C19.csv  GSM1658202_nochipID9.C92.csv\r\n",
      "GSM1657970_1772078237.C21.csv  GSM1658203_nochipID10.C02.csv\r\n",
      "GSM1657971_1772078237.C26.csv  GSM1658204_nochipID10.C03.csv\r\n",
      "GSM1657972_1772078237.C28.csv  GSM1658205_nochipID10.C05.csv\r\n",
      "GSM1657973_1772078237.C29.csv  GSM1658206_nochipID10.C07.csv\r\n",
      "GSM1657974_1772078237.C31.csv  GSM1658207_nochipID10.C11.csv\r\n",
      "GSM1657975_1772078237.C40.csv  GSM1658208_nochipID10.C16.csv\r\n",
      "GSM1657976_1772078237.C42.csv  GSM1658209_nochipID10.C17.csv\r\n",
      "GSM1657977_1772078237.C44.csv  GSM1658210_nochipID10.C19.csv\r\n",
      "GSM1657978_1772078237.C47.csv  GSM1658211_nochipID10.C25.csv\r\n",
      "GSM1657979_1772078237.C49.csv  GSM1658212_nochipID10.C26.csv\r\n",
      "GSM1657980_1772078237.C52.csv  GSM1658213_nochipID10.C37.csv\r\n",
      "GSM1657981_1772078237.C55.csv  GSM1658214_nochipID10.C39.csv\r\n",
      "GSM1657982_1772078237.C56.csv  GSM1658215_nochipID10.C42.csv\r\n",
      "GSM1657983_1772078237.C58.csv  GSM1658216_nochipID10.C44.csv\r\n",
      "GSM1657984_1772078237.C60.csv  GSM1658217_nochipID10.C47.csv\r\n",
      "GSM1657985_1772078237.C61.csv  GSM1658218_nochipID10.C49.csv\r\n",
      "GSM1657986_1772078237.C62.csv  GSM1658219_nochipID10.C50.csv\r\n",
      "GSM1657987_1772078237.C74.csv  GSM1658220_nochipID10.C53.csv\r\n",
      "GSM1657988_1772078237.C79.csv  GSM1658221_nochipID10.C58.csv\r\n",
      "GSM1657989_1772078237.C83.csv  GSM1658222_nochipID10.C65.csv\r\n",
      "GSM1657990_1772078237.C84.csv  GSM1658223_nochipID10.C74.csv\r\n",
      "GSM1657991_1772078237.C92.csv  GSM1658224_nochipID10.C75.csv\r\n",
      "GSM1657992_nochipID12.C05.csv  GSM1658225_nochipID10.C85.csv\r\n",
      "GSM1657993_nochipID12.C23.csv  GSM1658226_nochipID10.C88.csv\r\n",
      "GSM1657994_nochipID12.C73.csv  GSM1658227_nochipID10.C93.csv\r\n",
      "GSM1657995_nochipID12.C84.csv  GSM1658228_nochipID10.C94.csv\r\n",
      "GSM1657996_nochipID12.C87.csv  GSM1658229_nochipID11.C02.csv\r\n",
      "GSM1657997_nochipID14.C08.csv  GSM1658230_nochipID11.C03.csv\r\n",
      "GSM1657998_nochipID14.C13.csv  GSM1658231_nochipID11.C06.csv\r\n",
      "GSM1657999_nochipID14.C29.csv  GSM1658232_nochipID11.C07.csv\r\n",
      "GSM1658000_nochipID14.C45.csv  GSM1658233_nochipID11.C08.csv\r\n",
      "GSM1658001_nochipID14.C89.csv  GSM1658234_nochipID11.C10.csv\r\n",
      "GSM1658002_nochipID15.C20.csv  GSM1658235_nochipID11.C15.csv\r\n",
      "GSM1658003_nochipID15.C54.csv  GSM1658236_nochipID11.C17.csv\r\n",
      "GSM1658004_nochipID15.C69.csv  GSM1658237_nochipID11.C19.csv\r\n",
      "GSM1658005_nochipID15.C86.csv  GSM1658238_nochipID11.C20.csv\r\n",
      "GSM1658006_nochipID2.C01.csv   GSM1658239_nochipID11.C21.csv\r\n",
      "GSM1658007_nochipID2.C02.csv   GSM1658240_nochipID11.C22.csv\r\n",
      "GSM1658008_nochipID2.C03.csv   GSM1658241_nochipID11.C23.csv\r\n",
      "GSM1658009_nochipID2.C07.csv   GSM1658242_nochipID11.C25.csv\r\n",
      "GSM1658010_nochipID2.C08.csv   GSM1658243_nochipID11.C30.csv\r\n",
      "GSM1658011_nochipID2.C09.csv   GSM1658244_nochipID11.C37.csv\r\n",
      "GSM1658012_nochipID2.C10.csv   GSM1658245_nochipID11.C48.csv\r\n",
      "GSM1658013_nochipID2.C11.csv   GSM1658246_nochipID11.C49.csv\r\n",
      "GSM1658014_nochipID2.C12.csv   GSM1658247_nochipID11.C50.csv\r\n",
      "GSM1658015_nochipID2.C13.csv   GSM1658248_nochipID11.C53.csv\r\n",
      "GSM1658016_nochipID2.C14.csv   GSM1658249_nochipID11.C54.csv\r\n",
      "GSM1658017_nochipID2.C15.csv   GSM1658251_nochipID11.C55.csv\r\n",
      "GSM1658018_nochipID2.C16.csv   GSM1658253_nochipID11.C56.csv\r\n",
      "GSM1658019_nochipID2.C17.csv   GSM1658255_nochipID11.C57.csv\r\n",
      "GSM1658020_nochipID2.C18.csv   GSM1658257_nochipID11.C58.csv\r\n",
      "GSM1658021_nochipID2.C19.csv   GSM1658259_nochipID11.C59.csv\r\n",
      "GSM1658022_nochipID2.C21.csv   GSM1658262_nochipID11.C60.csv\r\n",
      "GSM1658023_nochipID2.C22.csv   GSM1658264_nochipID11.C61.csv\r\n",
      "GSM1658024_nochipID2.C24.csv   GSM1658266_nochipID11.C62.csv\r\n",
      "GSM1658025_nochipID2.C26.csv   GSM1658268_nochipID11.C63.csv\r\n",
      "GSM1658026_nochipID2.C27.csv   GSM1658270_nochipID11.C64.csv\r\n",
      "GSM1658027_nochipID2.C28.csv   GSM1658272_nochipID11.C66.csv\r\n",
      "GSM1658028_nochipID2.C29.csv   GSM1658275_nochipID11.C68.csv\r\n",
      "GSM1658029_nochipID2.C30.csv   GSM1658277_nochipID11.C69.csv\r\n",
      "GSM1658030_nochipID2.C31.csv   GSM1658279_nochipID11.C75.csv\r\n",
      "GSM1658031_nochipID2.C32.csv   GSM1658281_nochipID11.C76.csv\r\n",
      "GSM1658032_nochipID2.C33.csv   GSM1658284_nochipID11.C77.csv\r\n",
      "GSM1658033_nochipID2.C34.csv   GSM1658286_nochipID11.C78.csv\r\n",
      "GSM1658034_nochipID2.C36.csv   GSM1658288_nochipID11.C79.csv\r\n",
      "GSM1658035_nochipID2.C38.csv   GSM1658290_nochipID11.C81.csv\r\n",
      "GSM1658036_nochipID2.C39.csv   GSM1658292_nochipID11.C82.csv\r\n",
      "GSM1658037_nochipID2.C40.csv   GSM1658294_nochipID11.C83.csv\r\n",
      "GSM1658038_nochipID2.C41.csv   GSM1658297_nochipID11.C84.csv\r\n",
      "GSM1658039_nochipID2.C42.csv   GSM1658299_nochipID11.C86.csv\r\n",
      "GSM1658040_nochipID2.C43.csv   GSM1658301_nochipID11.C91.csv\r\n",
      "GSM1658041_nochipID2.C44.csv   GSM1658304_nochipID11.C93.csv\r\n",
      "GSM1658042_nochipID2.C45.csv   GSM1658305_nochipID13.C07.csv\r\n",
      "GSM1658043_nochipID2.C46.csv   GSM1658306_nochipID13.C11.csv\r\n",
      "GSM1658044_nochipID2.C47.csv   GSM1658307_nochipID13.C12.csv\r\n",
      "GSM1658045_nochipID2.C48.csv   GSM1658308_nochipID13.C14.csv\r\n",
      "GSM1658046_nochipID2.C49.csv   GSM1658309_nochipID13.C15.csv\r\n",
      "GSM1658047_nochipID2.C50.csv   GSM1658310_nochipID13.C17.csv\r\n",
      "GSM1658048_nochipID2.C51.csv   GSM1658311_nochipID13.C21.csv\r\n",
      "GSM1658049_nochipID2.C52.csv   GSM1658312_nochipID13.C25.csv\r\n",
      "GSM1658050_nochipID2.C56.csv   GSM1658313_nochipID13.C28.csv\r\n",
      "GSM1658051_nochipID2.C57.csv   GSM1658314_nochipID13.C32.csv\r\n",
      "GSM1658052_nochipID2.C58.csv   GSM1658315_nochipID13.C36.csv\r\n",
      "GSM1658053_nochipID2.C59.csv   GSM1658316_nochipID13.C45.csv\r\n",
      "GSM1658054_nochipID2.C60.csv   GSM1658317_nochipID13.C46.csv\r\n",
      "GSM1658055_nochipID2.C61.csv   GSM1658318_nochipID13.C48.csv\r\n",
      "GSM1658056_nochipID2.C62.csv   GSM1658319_nochipID13.C50.csv\r\n",
      "GSM1658057_nochipID2.C63.csv   GSM1658320_nochipID13.C55.csv\r\n",
      "GSM1658058_nochipID2.C64.csv   GSM1658321_nochipID13.C57.csv\r\n",
      "GSM1658059_nochipID2.C65.csv   GSM1658322_nochipID13.C58.csv\r\n",
      "GSM1658060_nochipID2.C66.csv   GSM1658323_nochipID13.C59.csv\r\n",
      "GSM1658061_nochipID2.C67.csv   GSM1658324_nochipID13.C61.csv\r\n",
      "GSM1658062_nochipID2.C68.csv   GSM1658325_nochipID13.C63.csv\r\n",
      "GSM1658063_nochipID2.C69.csv   GSM1658326_nochipID13.C64.csv\r\n",
      "GSM1658064_nochipID2.C71.csv   GSM1658327_nochipID13.C66.csv\r\n",
      "GSM1658065_nochipID2.C73.csv   GSM1658328_nochipID13.C67.csv\r\n",
      "GSM1658066_nochipID2.C75.csv   GSM1658329_nochipID13.C68.csv\r\n",
      "GSM1658067_nochipID2.C76.csv   GSM1658330_nochipID13.C72.csv\r\n",
      "GSM1658068_nochipID2.C77.csv   GSM1658331_nochipID13.C73.csv\r\n",
      "GSM1658069_nochipID2.C78.csv   GSM1658332_nochipID13.C74.csv\r\n",
      "GSM1658070_nochipID2.C79.csv   GSM1658333_nochipID13.C75.csv\r\n",
      "GSM1658071_nochipID2.C80.csv   GSM1658334_nochipID13.C80.csv\r\n",
      "GSM1658072_nochipID2.C81.csv   GSM1658335_nochipID13.C84.csv\r\n",
      "GSM1658073_nochipID2.C82.csv   GSM1658336_nochipID13.C87.csv\r\n",
      "GSM1658074_nochipID2.C83.csv   GSM1658337_nochipID13.C93.csv\r\n",
      "GSM1658075_nochipID2.C84.csv   GSM1658338_nochipID4.C03.csv\r\n",
      "GSM1658076_nochipID2.C85.csv   GSM1658339_nochipID4.C04.csv\r\n",
      "GSM1658077_nochipID2.C86.csv   GSM1658340_nochipID4.C10.csv\r\n",
      "GSM1658078_nochipID2.C87.csv   GSM1658341_nochipID4.C14.csv\r\n",
      "GSM1658079_nochipID2.C89.csv   GSM1658342_nochipID4.C18.csv\r\n",
      "GSM1658080_nochipID2.C90.csv   GSM1658343_nochipID4.C19.csv\r\n",
      "GSM1658081_nochipID2.C92.csv   GSM1658344_nochipID4.C20.csv\r\n",
      "GSM1658082_nochipID2.C93.csv   GSM1658345_nochipID4.C21.csv\r\n",
      "GSM1658083_nochipID3.C01.csv   GSM1658346_nochipID4.C33.csv\r\n",
      "GSM1658084_nochipID3.C05.csv   GSM1658347_nochipID4.C34.csv\r\n",
      "GSM1658085_nochipID3.C06.csv   GSM1658348_nochipID4.C38.csv\r\n",
      "GSM1658086_nochipID3.C07.csv   GSM1658349_nochipID4.C39.csv\r\n",
      "GSM1658087_nochipID3.C09.csv   GSM1658350_nochipID4.C41.csv\r\n",
      "GSM1658088_nochipID3.C11.csv   GSM1658351_nochipID4.C42.csv\r\n",
      "GSM1658089_nochipID3.C12.csv   GSM1658352_nochipID4.C44.csv\r\n",
      "GSM1658090_nochipID3.C14.csv   GSM1658353_nochipID4.C49.csv\r\n",
      "GSM1658091_nochipID3.C23.csv   GSM1658354_nochipID4.C52.csv\r\n",
      "GSM1658092_nochipID3.C25.csv   GSM1658355_nochipID4.C53.csv\r\n",
      "GSM1658093_nochipID3.C32.csv   GSM1658356_nochipID4.C59.csv\r\n",
      "GSM1658094_nochipID3.C37.csv   GSM1658357_nochipID4.C62.csv\r\n",
      "GSM1658095_nochipID3.C38.csv   GSM1658358_nochipID4.C63.csv\r\n",
      "GSM1658096_nochipID3.C51.csv   GSM1658359_nochipID4.C66.csv\r\n",
      "GSM1658097_nochipID3.C52.csv   GSM1658360_nochipID4.C74.csv\r\n",
      "GSM1658098_nochipID3.C53.csv   GSM1658361_nochipID4.C77.csv\r\n",
      "GSM1658099_nochipID3.C54.csv   GSM1658362_nochipID4.C78.csv\r\n",
      "GSM1658100_nochipID3.C56.csv   GSM1658363_nochipID4.C84.csv\r\n",
      "GSM1658101_nochipID3.C61.csv   GSM1658364_nochipID4.C89.csv\r\n",
      "GSM1658102_nochipID3.C62.csv   GSM1658365_nochipID4.C95.csv\r\n",
      "GSM1658103_nochipID3.C65.csv   GSM1658366_nochipID4.C96.csv\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1/2-SBSRNA4 \t 0\r\n",
      "A1BG \t 0\r\n",
      "A1BG-AS1 \t 0\r\n",
      "A1CF \t 0\r\n",
      "A2LD1 \t 0\r\n",
      "A2M \t 18\r\n",
      "A2ML1 \t 0\r\n",
      "A2MP1 \t 0\r\n",
      "A4GALT \t 0\r\n",
      "A4GNT \t 0\r\n"
     ]
    }
   ],
   "source": [
    "! head GSM1657872_1772078217.C04.csv "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import glob\n",
    "\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1/2-SBSRNA4</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2LD1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>A2MP1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>A4GNT</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AA06</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AAA1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AACS</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AACSP1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AADAC</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AADACL2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AADACL3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>AADACL4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>AAMP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AANAT</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>AARS</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>AARSD1</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>AASDH</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>AASDHPPT</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>AASS</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>AATF</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22057</th>\n",
       "      <td>ZSCAN22</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22058</th>\n",
       "      <td>ZSCAN23</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22059</th>\n",
       "      <td>ZSCAN29</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22060</th>\n",
       "      <td>ZSCAN30</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22061</th>\n",
       "      <td>ZSCAN4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22062</th>\n",
       "      <td>ZSCAN5A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22063</th>\n",
       "      <td>ZSCAN5B</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22064</th>\n",
       "      <td>ZSWIM1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22065</th>\n",
       "      <td>ZSWIM2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22066</th>\n",
       "      <td>ZSWIM3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22067</th>\n",
       "      <td>ZSWIM4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22068</th>\n",
       "      <td>ZSWIM5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22069</th>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22070</th>\n",
       "      <td>ZSWIM7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22071</th>\n",
       "      <td>ZUFSP</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22072</th>\n",
       "      <td>ZW10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22073</th>\n",
       "      <td>ZWILCH</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22074</th>\n",
       "      <td>ZWINT</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22075</th>\n",
       "      <td>ZXDA</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22076</th>\n",
       "      <td>ZXDB</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22077</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22078</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22079</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22080</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22081</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22082</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22083</th>\n",
       "      <td>tAKR</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22084</th>\n",
       "      <td>no_feature</td>\n",
       "      <td>970221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22085</th>\n",
       "      <td>ambiguous</td>\n",
       "      <td>5771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22086</th>\n",
       "      <td>alignment_not_unique</td>\n",
       "      <td>847047</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22087 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                1/2-SBSRNA4        0\n",
       "0                      A1BG        0\n",
       "1                  A1BG-AS1        0\n",
       "2                      A1CF        0\n",
       "3                     A2LD1        0\n",
       "4                       A2M       18\n",
       "5                     A2ML1        0\n",
       "6                     A2MP1        0\n",
       "7                    A4GALT        0\n",
       "8                     A4GNT        0\n",
       "9                      AA06        0\n",
       "10                     AAA1        0\n",
       "11                     AAAS        0\n",
       "12                     AACS        0\n",
       "13                   AACSP1        0\n",
       "14                    AADAC        0\n",
       "15                  AADACL2        0\n",
       "16                  AADACL3        0\n",
       "17                  AADACL4        0\n",
       "18                    AADAT        0\n",
       "19                    AAGAB      572\n",
       "20                     AAK1      174\n",
       "21                     AAMP        1\n",
       "22                    AANAT        0\n",
       "23                     AARS        8\n",
       "24                    AARS2        0\n",
       "25                   AARSD1       57\n",
       "26                    AASDH        0\n",
       "27                 AASDHPPT        0\n",
       "28                     AASS        0\n",
       "29                     AATF        0\n",
       "...                      ...     ...\n",
       "22057               ZSCAN22      285\n",
       "22058               ZSCAN23        0\n",
       "22059               ZSCAN29        4\n",
       "22060               ZSCAN30        0\n",
       "22061                ZSCAN4        0\n",
       "22062               ZSCAN5A        0\n",
       "22063               ZSCAN5B        0\n",
       "22064                ZSWIM1        0\n",
       "22065                ZSWIM2        0\n",
       "22066                ZSWIM3        0\n",
       "22067                ZSWIM4        0\n",
       "22068                ZSWIM5        0\n",
       "22069                ZSWIM6       69\n",
       "22070                ZSWIM7        0\n",
       "22071                 ZUFSP        0\n",
       "22072                  ZW10        0\n",
       "22073                ZWILCH        2\n",
       "22074                 ZWINT        0\n",
       "22075                  ZXDA        0\n",
       "22076                  ZXDB        0\n",
       "22077                  ZXDC        5\n",
       "22078                ZYG11A        0\n",
       "22079                ZYG11B      197\n",
       "22080                   ZYX        0\n",
       "22081                 ZZEF1        0\n",
       "22082                  ZZZ3       58\n",
       "22083                  tAKR        0\n",
       "22084            no_feature   970221\n",
       "22085             ambiguous     5771\n",
       "22086  alignment_not_unique   847047\n",
       "\n",
       "[22087 rows x 2 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_table('GSM1657872_1772078217.C04.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/2-SBSRNA4</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2LD1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2MP1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GNT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AA06</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAA1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACSP1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADAC</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADACL2</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADACL3</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADACL4</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADAT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAGAB</th>\n",
       "      <td>572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAK1</th>\n",
       "      <td>174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAMP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AANAT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AARS</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AARS2</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AARSD1</th>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AASDH</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AASDHPPT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AASS</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AATF</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN22</th>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN23</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN29</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN30</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN4</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN5A</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN5B</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM2</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM3</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM4</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM5</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM6</th>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM7</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZUFSP</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZW10</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZWILCH</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZWINT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDA</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDB</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11A</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tAKR</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no_feature</th>\n",
       "      <td>970221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ambiguous</th>\n",
       "      <td>5771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>alignment_not_unique</th>\n",
       "      <td>847047</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22087 rows  1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            0\n",
       "1/2-SBSRNA4                  \n",
       "A1BG                        0\n",
       "A1BG-AS1                    0\n",
       "A1CF                        0\n",
       "A2LD1                       0\n",
       "A2M                        18\n",
       "A2ML1                       0\n",
       "A2MP1                       0\n",
       "A4GALT                      0\n",
       "A4GNT                       0\n",
       "AA06                        0\n",
       "AAA1                        0\n",
       "AAAS                        0\n",
       "AACS                        0\n",
       "AACSP1                      0\n",
       "AADAC                       0\n",
       "AADACL2                     0\n",
       "AADACL3                     0\n",
       "AADACL4                     0\n",
       "AADAT                       0\n",
       "AAGAB                     572\n",
       "AAK1                      174\n",
       "AAMP                        1\n",
       "AANAT                       0\n",
       "AARS                        8\n",
       "AARS2                       0\n",
       "AARSD1                     57\n",
       "AASDH                       0\n",
       "AASDHPPT                    0\n",
       "AASS                        0\n",
       "AATF                        0\n",
       "...                       ...\n",
       "ZSCAN22                   285\n",
       "ZSCAN23                     0\n",
       "ZSCAN29                     4\n",
       "ZSCAN30                     0\n",
       "ZSCAN4                      0\n",
       "ZSCAN5A                     0\n",
       "ZSCAN5B                     0\n",
       "ZSWIM1                      0\n",
       "ZSWIM2                      0\n",
       "ZSWIM3                      0\n",
       "ZSWIM4                      0\n",
       "ZSWIM5                      0\n",
       "ZSWIM6                     69\n",
       "ZSWIM7                      0\n",
       "ZUFSP                       0\n",
       "ZW10                        0\n",
       "ZWILCH                      2\n",
       "ZWINT                       0\n",
       "ZXDA                        0\n",
       "ZXDB                        0\n",
       "ZXDC                        5\n",
       "ZYG11A                      0\n",
       "ZYG11B                    197\n",
       "ZYX                         0\n",
       "ZZEF1                       0\n",
       "ZZZ3                       58\n",
       "tAKR                        0\n",
       "no_feature             970221\n",
       "ambiguous                5771\n",
       "alignment_not_unique   847047\n",
       "\n",
       "[22087 rows x 1 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_table('GSM1657872_1772078217.C04.csv', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1/2-SBSRNA4</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2LD1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2MP1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GNT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AA06</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAA1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACSP1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADAC</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADACL2</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADACL3</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADACL4</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADAT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAGAB</th>\n",
       "      <td>572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAK1</th>\n",
       "      <td>174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAMP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AANAT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AARS</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AARS2</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AARSD1</th>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AASDH</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AASDHPPT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AASS</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN22</th>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN23</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN29</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN30</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN4</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN5A</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN5B</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM2</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM3</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM4</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM5</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM6</th>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM7</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZUFSP</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZW10</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZWILCH</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZWINT</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDA</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDB</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11A</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tAKR</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no_feature</th>\n",
       "      <td>970221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ambiguous</th>\n",
       "      <td>5771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>alignment_not_unique</th>\n",
       "      <td>847047</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22088 rows  1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            1\n",
       "0                            \n",
       "1/2-SBSRNA4                 0\n",
       "A1BG                        0\n",
       "A1BG-AS1                    0\n",
       "A1CF                        0\n",
       "A2LD1                       0\n",
       "A2M                        18\n",
       "A2ML1                       0\n",
       "A2MP1                       0\n",
       "A4GALT                      0\n",
       "A4GNT                       0\n",
       "AA06                        0\n",
       "AAA1                        0\n",
       "AAAS                        0\n",
       "AACS                        0\n",
       "AACSP1                      0\n",
       "AADAC                       0\n",
       "AADACL2                     0\n",
       "AADACL3                     0\n",
       "AADACL4                     0\n",
       "AADAT                       0\n",
       "AAGAB                     572\n",
       "AAK1                      174\n",
       "AAMP                        1\n",
       "AANAT                       0\n",
       "AARS                        8\n",
       "AARS2                       0\n",
       "AARSD1                     57\n",
       "AASDH                       0\n",
       "AASDHPPT                    0\n",
       "AASS                        0\n",
       "...                       ...\n",
       "ZSCAN22                   285\n",
       "ZSCAN23                     0\n",
       "ZSCAN29                     4\n",
       "ZSCAN30                     0\n",
       "ZSCAN4                      0\n",
       "ZSCAN5A                     0\n",
       "ZSCAN5B                     0\n",
       "ZSWIM1                      0\n",
       "ZSWIM2                      0\n",
       "ZSWIM3                      0\n",
       "ZSWIM4                      0\n",
       "ZSWIM5                      0\n",
       "ZSWIM6                     69\n",
       "ZSWIM7                      0\n",
       "ZUFSP                       0\n",
       "ZW10                        0\n",
       "ZWILCH                      2\n",
       "ZWINT                       0\n",
       "ZXDA                        0\n",
       "ZXDB                        0\n",
       "ZXDC                        5\n",
       "ZYG11A                      0\n",
       "ZYG11B                    197\n",
       "ZYX                         0\n",
       "ZZEF1                       0\n",
       "ZZZ3                       58\n",
       "tAKR                        0\n",
       "no_feature             970221\n",
       "ambiguous                5771\n",
       "alignment_not_unique   847047\n",
       "\n",
       "[22088 rows x 1 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataframe = pd.read_table('GSM1657872_1772078217.C04.csv', index_col=0, header=None)\n",
    "dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0\n",
       "1/2-SBSRNA4                   0\n",
       "A1BG                          0\n",
       "A1BG-AS1                      0\n",
       "A1CF                          0\n",
       "A2LD1                         0\n",
       "A2M                          18\n",
       "A2ML1                         0\n",
       "A2MP1                         0\n",
       "A4GALT                        0\n",
       "A4GNT                         0\n",
       "AA06                          0\n",
       "AAA1                          0\n",
       "AAAS                          0\n",
       "AACS                          0\n",
       "AACSP1                        0\n",
       "AADAC                         0\n",
       "AADACL2                       0\n",
       "AADACL3                       0\n",
       "AADACL4                       0\n",
       "AADAT                         0\n",
       "AAGAB                       572\n",
       "AAK1                        174\n",
       "AAMP                          1\n",
       "AANAT                         0\n",
       "AARS                          8\n",
       "AARS2                         0\n",
       "AARSD1                       57\n",
       "AASDH                         0\n",
       "AASDHPPT                      0\n",
       "AASS                          0\n",
       "                          ...  \n",
       "ZSCAN22                     285\n",
       "ZSCAN23                       0\n",
       "ZSCAN29                       4\n",
       "ZSCAN30                       0\n",
       "ZSCAN4                        0\n",
       "ZSCAN5A                       0\n",
       "ZSCAN5B                       0\n",
       "ZSWIM1                        0\n",
       "ZSWIM2                        0\n",
       "ZSWIM3                        0\n",
       "ZSWIM4                        0\n",
       "ZSWIM5                        0\n",
       "ZSWIM6                       69\n",
       "ZSWIM7                        0\n",
       "ZUFSP                         0\n",
       "ZW10                          0\n",
       "ZWILCH                        2\n",
       "ZWINT                         0\n",
       "ZXDA                          0\n",
       "ZXDB                          0\n",
       "ZXDC                          5\n",
       "ZYG11A                        0\n",
       "ZYG11B                      197\n",
       "ZYX                           0\n",
       "ZZEF1                         0\n",
       "ZZZ3                         58\n",
       "tAKR                          0\n",
       "no_feature               970221\n",
       "ambiguous                  5771\n",
       "alignment_not_unique     847047\n",
       "Name: 1, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "series = pd.read_table('GSM1657872_1772078217.C04.csv', index_col=0, header=None, squeeze=True)\n",
    "series"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22088, 1)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataframe.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22088,)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "series.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "series.name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GSM1657872_1772078217.C04.csv'"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filename = 'GSM1657872_1772078217.C04.csv'\n",
    "filename"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GSM1657872_1772078217', 'C04', 'csv']"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filename.split('.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GSM1657872_1772078217.C04', '']"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filename.split('.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GSM1657872_1772078217.C04'"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filename.split('.csv')[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(22088, 466)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSM1657871</th>\n",
       "      <th>GSM1657872</th>\n",
       "      <th>GSM1657873</th>\n",
       "      <th>GSM1657874</th>\n",
       "      <th>GSM1657875</th>\n",
       "      <th>GSM1657876</th>\n",
       "      <th>GSM1657877</th>\n",
       "      <th>GSM1657878</th>\n",
       "      <th>GSM1657879</th>\n",
       "      <th>GSM1657880</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM1658357</th>\n",
       "      <th>GSM1658358</th>\n",
       "      <th>GSM1658359</th>\n",
       "      <th>GSM1658360</th>\n",
       "      <th>GSM1658361</th>\n",
       "      <th>GSM1658362</th>\n",
       "      <th>GSM1658363</th>\n",
       "      <th>GSM1658364</th>\n",
       "      <th>GSM1658365</th>\n",
       "      <th>GSM1658366</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1/2-SBSRNA4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2LD1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  466 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             GSM1657871  GSM1657872  GSM1657873  GSM1657874  GSM1657875  \\\n",
       "1/2-SBSRNA4           0           0           0           0           0   \n",
       "A1BG                  0           0           0           0           0   \n",
       "A1BG-AS1              0           0           0           0           0   \n",
       "A1CF                  0           0           0           0           0   \n",
       "A2LD1                 0           0           0           0           0   \n",
       "\n",
       "             GSM1657876  GSM1657877  GSM1657878  GSM1657879  GSM1657880  \\\n",
       "1/2-SBSRNA4           0           0           0           0           0   \n",
       "A1BG                  0           0           0           0           0   \n",
       "A1BG-AS1              0           0           0           0           0   \n",
       "A1CF                  0           0           0           0           0   \n",
       "A2LD1                 0           0           0           0           0   \n",
       "\n",
       "                ...      GSM1658357  GSM1658358  GSM1658359  GSM1658360  \\\n",
       "1/2-SBSRNA4     ...               0           0           0           0   \n",
       "A1BG            ...               0           0           0           0   \n",
       "A1BG-AS1        ...               0           0           0           0   \n",
       "A1CF            ...               0           0           0           0   \n",
       "A2LD1           ...               0           0           0           0   \n",
       "\n",
       "             GSM1658361  GSM1658362  GSM1658363  GSM1658364  GSM1658365  \\\n",
       "1/2-SBSRNA4           0           0           0           0           0   \n",
       "A1BG                  0           0           0           0           0   \n",
       "A1BG-AS1              0           0           0           0           0   \n",
       "A1CF                  0           0           0           0           0   \n",
       "A2LD1                 0           0           0           0           0   \n",
       "\n",
       "             GSM1658366  \n",
       "1/2-SBSRNA4           0  \n",
       "A1BG                  0  \n",
       "A1BG-AS1              0  \n",
       "A1CF                  0  \n",
       "A2LD1                 0  \n",
       "\n",
       "[5 rows x 466 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cells = []\n",
    "\n",
    "for filename in glob.iglob('*.csv'):\n",
    "    cell = pd.read_table(filename, index_col=0, squeeze=True, header=None)\n",
    "    name = filename.split('_')[0]\n",
    "    cell.name = name\n",
    "    cells.append(cell)\n",
    "expression = pd.concat(cells, axis=1)\n",
    "expression.index = expression.index.map(lambda x: x.strip(' '))\n",
    "print(expression.shape)\n",
    "expression.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "! gunzip /Users/kirkreardon/Downloads/*_series_matrix.txt.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /Users/kirkreardon/Downloads/GSE67835-GPL15520_series_matrix.txt <==\r\n",
      "!Series_title\t\"A survey of human brain transcriptome diversity at the single cell level\"\r\n",
      "!Series_geo_accession\t\"GSE67835\"\r\n",
      "!Series_status\t\"Public on May 20 2015\"\r\n",
      "!Series_submission_date\t\"Apr 14 2015\"\r\n",
      "!Series_last_update_date\t\"May 26 2016\"\r\n",
      "!Series_pubmed_id\t\"26060301\"\r\n",
      "!Series_summary\t\"We used single cell RNA sequencing on 466 cells to capture the cellular complexity of the adult and fetal human brain at a whole transcriptome level. Healthy adult temporal lobe tissue was obtained from epileptic patients during temporal lobectomy for medically refractory seizures. We were able to classify individual cells into all of the major neuronal, glial, and vascular cell types in the brain.\"\r\n",
      "!Series_overall_design\t\"Examination of cell types in healthy human brain samples.\"\r\n",
      "!Series_type\t\"Expression profiling by high throughput sequencing\"\r\n",
      "!Series_contributor\t\"Spyros,,Darmanis\"\r\n",
      "\r\n",
      "==> /Users/kirkreardon/Downloads/GSE67835-GPL18573_series_matrix.txt <==\r\n",
      "!Series_title\t\"A survey of human brain transcriptome diversity at the single cell level\"\r\n",
      "!Series_geo_accession\t\"GSE67835\"\r\n",
      "!Series_status\t\"Public on May 20 2015\"\r\n",
      "!Series_submission_date\t\"Apr 14 2015\"\r\n",
      "!Series_last_update_date\t\"May 26 2016\"\r\n",
      "!Series_pubmed_id\t\"26060301\"\r\n",
      "!Series_summary\t\"We used single cell RNA sequencing on 466 cells to capture the cellular complexity of the adult and fetal human brain at a whole transcriptome level. Healthy adult temporal lobe tissue was obtained from epileptic patients during temporal lobectomy for medically refractory seizures. We were able to classify individual cells into all of the major neuronal, glial, and vascular cell types in the brain.\"\r\n",
      "!Series_overall_design\t\"Examination of cell types in healthy human brain samples.\"\r\n",
      "!Series_type\t\"Expression profiling by high throughput sequencing\"\r\n",
      "!Series_contributor\t\"Spyros,,Darmanis\"\r\n"
     ]
    }
   ],
   "source": [
    "! head /Users/kirkreardon/Downloads/*_series_matrix.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /Users/kirkreardon/Downloads/GSE67835-GPL15520_series_matrix.txt <==\r\n",
      "!Series_title\t\"A survey of human brain transcriptome diversity at the single cell level\"\r\n",
      "!Series_geo_accession\t\"GSE67835\"\r\n",
      "!Series_status\t\"Public on May 20 2015\"\r\n",
      "!Series_submission_date\t\"Apr 14 2015\"\r\n",
      "!Series_last_update_date\t\"May 26 2016\"\r\n",
      "!Series_pubmed_id\t\"26060301\"\r\n",
      "!Series_summary\t\"We used single cell RNA sequencing on 466 cells to capture the cellular complexity of the adult and fetal human brain at a whole transcriptome level. Healthy adult temporal lobe tissue was obtained from epileptic patients during temporal lobectomy for medically refractory seizures. We were able to classify individual cells into all of the major neuronal, glial, and vascular cell types in the brain.\"\r\n",
      "!Series_overall_design\t\"Examination of cell types in healthy human brain samples.\"\r\n",
      "!Series_type\t\"Expression profiling by high throughput sequencing\"\r\n",
      "!Series_contributor\t\"Spyros,,Darmanis\"\r\n",
      "!Series_contributor\t\"Martin,,Enge\"\r\n",
      "!Series_contributor\t\"Stephen,R,Quake\"\r\n",
      "!Series_contributor\t\"Steven,A,Sloan\"\r\n",
      "!Series_contributor\t\"Ben,A,Barres\"\r\n",
      "!Series_contributor\t\"Ye,,Zhang\"\r\n",
      "!Series_contributor\t\"Christine,,Caneda\"\r\n",
      "!Series_contributor\t\"Melanie,G,Hayden Gephart\"\r\n",
      "!Series_contributor\t\"Lawrence,M,Shuer\"\r\n",
      "!Series_sample_id\t\"GSM1657871 GSM1657872 GSM1657873 GSM1657874 GSM1657875 GSM1657876 GSM1657877 GSM1657878 GSM1657879 GSM1657880 GSM1657881 GSM1657882 GSM1657883 GSM1657884 GSM1657885 GSM1657886 GSM1657887 GSM1657888 GSM1657889 GSM1657890 GSM1657891 GSM1657892 GSM1657893 GSM1657894 GSM1657895 GSM1657896 GSM1657897 GSM1657898 GSM1657899 GSM1657900 GSM1657901 GSM1657902 GSM1657903 GSM1657904 GSM1657905 GSM1657906 GSM1657907 GSM1657908 GSM1657909 GSM1657910 GSM1657911 GSM1657912 GSM1657913 GSM1657914 GSM1657915 GSM1657916 GSM1657917 GSM1657918 GSM1657919 GSM1657920 GSM1657921 GSM1657922 GSM1657923 GSM1657924 GSM1657925 GSM1657926 GSM1657927 GSM1657928 GSM1657929 GSM1657930 GSM1657931 GSM1657932 GSM1657933 GSM1657934 GSM1657935 GSM1657936 GSM1657937 GSM1657938 GSM1657939 GSM1657940 GSM1657941 GSM1657942 GSM1657943 GSM1657944 GSM1657945 GSM1657946 GSM1657947 GSM1657948 GSM1657949 GSM1657950 GSM1657951 GSM1657952 GSM1657953 GSM1657954 GSM1657955 GSM1657956 GSM1657957 GSM1657958 GSM1657959 GSM1657960 GSM1657961 GSM1657962 GSM1657963 GSM1657964 GSM1657965 GSM1657966 GSM1657967 GSM1657968 GSM1657969 GSM1657970 GSM1657971 GSM1657972 GSM1657973 GSM1657974 GSM1657975 GSM1657976 GSM1657977 GSM1657978 GSM1657979 GSM1657980 GSM1657981 GSM1657982 GSM1657983 GSM1657984 GSM1657985 GSM1657986 GSM1657987 GSM1657988 GSM1657989 GSM1657990 GSM1657991 GSM1657992 GSM1657993 GSM1657994 GSM1657995 GSM1657996 GSM1657997 GSM1657998 GSM1657999 GSM1658000 GSM1658001 GSM1658002 GSM1658003 GSM1658004 GSM1658005 GSM1658006 GSM1658007 GSM1658008 GSM1658009 GSM1658010 GSM1658011 GSM1658012 GSM1658013 GSM1658014 GSM1658015 GSM1658016 GSM1658017 GSM1658018 GSM1658019 GSM1658020 GSM1658021 GSM1658022 GSM1658023 GSM1658024 GSM1658025 GSM1658026 GSM1658027 GSM1658028 GSM1658029 GSM1658030 GSM1658031 GSM1658032 GSM1658033 GSM1658034 GSM1658035 GSM1658036 GSM1658037 GSM1658038 GSM1658039 GSM1658040 GSM1658041 GSM1658042 GSM1658043 GSM1658044 GSM1658045 GSM1658046 GSM1658047 GSM1658048 GSM1658049 GSM1658050 GSM1658051 GSM1658052 GSM1658053 GSM1658054 GSM1658055 GSM1658056 GSM1658057 GSM1658058 GSM1658059 GSM1658060 GSM1658061 GSM1658062 GSM1658063 GSM1658064 GSM1658065 GSM1658066 GSM1658067 GSM1658068 GSM1658069 GSM1658070 GSM1658071 GSM1658072 GSM1658073 GSM1658074 GSM1658075 GSM1658076 GSM1658077 GSM1658078 GSM1658079 GSM1658080 GSM1658081 GSM1658082 GSM1658083 GSM1658084 GSM1658085 GSM1658086 GSM1658087 GSM1658088 GSM1658089 GSM1658090 GSM1658091 GSM1658092 GSM1658093 GSM1658094 GSM1658095 GSM1658096 GSM1658097 GSM1658098 GSM1658099 GSM1658100 GSM1658101 GSM1658102 GSM1658103 GSM1658104 GSM1658105 GSM1658106 GSM1658107 GSM1658108 GSM1658109 GSM1658110 GSM1658111 GSM1658112 GSM1658113 GSM1658114 GSM1658115 GSM1658116 GSM1658117 GSM1658118 GSM1658119 GSM1658120 GSM1658121 GSM1658122 GSM1658123 GSM1658124 GSM1658125 GSM1658126 GSM1658127 GSM1658128 GSM1658129 GSM1658130 GSM1658131 GSM1658132 GSM1658133 GSM1658134 GSM1658135 GSM1658136 GSM1658137 GSM1658138 GSM1658139 GSM1658140 GSM1658141 GSM1658142 GSM1658143 GSM1658144 GSM1658145 GSM1658146 GSM1658147 GSM1658148 GSM1658149 GSM1658150 GSM1658151 GSM1658152 GSM1658153 GSM1658154 GSM1658155 GSM1658156 GSM1658157 GSM1658158 GSM1658159 GSM1658160 GSM1658161 GSM1658162 GSM1658163 GSM1658164 GSM1658165 GSM1658166 GSM1658167 GSM1658168 GSM1658169 GSM1658170 GSM1658171 GSM1658172 GSM1658173 GSM1658174 GSM1658175 GSM1658176 GSM1658177 GSM1658178 GSM1658179 GSM1658180 GSM1658181 GSM1658182 GSM1658183 GSM1658184 GSM1658185 GSM1658186 GSM1658187 GSM1658188 GSM1658189 GSM1658190 GSM1658191 GSM1658192 GSM1658193 GSM1658194 GSM1658195 GSM1658196 GSM1658197 GSM1658198 GSM1658199 GSM1658200 GSM1658201 GSM1658202 GSM1658203 GSM1658204 GSM1658205 GSM1658206 GSM1658207 GSM1658208 GSM1658209 GSM1658210 GSM1658211 GSM1658212 GSM1658213 GSM1658214 GSM1658215 GSM1658216 GSM1658217 GSM1658218 GSM1658219 GSM1658220 GSM1658221 GSM1658222 GSM1658223 GSM1658224 GSM1658225 GSM1658226 GSM1658227 GSM1658228 GSM1658229 GSM1658230 GSM1658231 GSM1658232 GSM1658233 GSM1658234 GSM1658235 GSM1658236 GSM1658237 GSM1658238 GSM1658239 GSM1658240 GSM1658241 GSM1658242 GSM1658243 GSM1658244 GSM1658245 GSM1658246 GSM1658247 GSM1658248 GSM1658249 GSM1658251 GSM1658253 GSM1658255 GSM1658257 GSM1658259 GSM1658262 GSM1658264 GSM1658266 GSM1658268 GSM1658270 GSM1658272 GSM1658275 GSM1658277 GSM1658279 GSM1658281 GSM1658284 GSM1658286 GSM1658288 GSM1658290 GSM1658292 GSM1658294 GSM1658297 GSM1658299 GSM1658301 GSM1658304 GSM1658305 GSM1658306 GSM1658307 GSM1658308 GSM1658309 GSM1658310 GSM1658311 GSM1658312 GSM1658313 GSM1658314 GSM1658315 GSM1658316 GSM1658317 GSM1658318 GSM1658319 GSM1658320 GSM1658321 GSM1658322 GSM1658323 GSM1658324 GSM1658325 GSM1658326 GSM1658327 GSM1658328 GSM1658329 GSM1658330 GSM1658331 GSM1658332 GSM1658333 GSM1658334 GSM1658335 GSM1658336 GSM1658337 GSM1658338 GSM1658339 GSM1658340 GSM1658341 GSM1658342 GSM1658343 GSM1658344 GSM1658345 GSM1658346 GSM1658347 GSM1658348 GSM1658349 GSM1658350 GSM1658351 GSM1658352 GSM1658353 GSM1658354 GSM1658355 GSM1658356 GSM1658357 GSM1658358 GSM1658359 GSM1658360 GSM1658361 GSM1658362 GSM1658363 GSM1658364 GSM1658365 GSM1658366 \"\r\n",
      "!Series_contact_name\t\"Martin,,Enge\"\r\n",
      "\r\n",
      "==> /Users/kirkreardon/Downloads/GSE67835-GPL18573_series_matrix.txt <==\r\n",
      "!Series_title\t\"A survey of human brain transcriptome diversity at the single cell level\"\r\n",
      "!Series_geo_accession\t\"GSE67835\"\r\n",
      "!Series_status\t\"Public on May 20 2015\"\r\n",
      "!Series_submission_date\t\"Apr 14 2015\"\r\n",
      "!Series_last_update_date\t\"May 26 2016\"\r\n",
      "!Series_pubmed_id\t\"26060301\"\r\n",
      "!Series_summary\t\"We used single cell RNA sequencing on 466 cells to capture the cellular complexity of the adult and fetal human brain at a whole transcriptome level. Healthy adult temporal lobe tissue was obtained from epileptic patients during temporal lobectomy for medically refractory seizures. We were able to classify individual cells into all of the major neuronal, glial, and vascular cell types in the brain.\"\r\n",
      "!Series_overall_design\t\"Examination of cell types in healthy human brain samples.\"\r\n",
      "!Series_type\t\"Expression profiling by high throughput sequencing\"\r\n",
      "!Series_contributor\t\"Spyros,,Darmanis\"\r\n",
      "!Series_contributor\t\"Martin,,Enge\"\r\n",
      "!Series_contributor\t\"Stephen,R,Quake\"\r\n",
      "!Series_contributor\t\"Steven,A,Sloan\"\r\n",
      "!Series_contributor\t\"Ben,A,Barres\"\r\n",
      "!Series_contributor\t\"Ye,,Zhang\"\r\n",
      "!Series_contributor\t\"Christine,,Caneda\"\r\n",
      "!Series_contributor\t\"Melanie,G,Hayden Gephart\"\r\n",
      "!Series_contributor\t\"Lawrence,M,Shuer\"\r\n",
      "!Series_sample_id\t\"GSM1657871 GSM1657872 GSM1657873 GSM1657874 GSM1657875 GSM1657876 GSM1657877 GSM1657878 GSM1657879 GSM1657880 GSM1657881 GSM1657882 GSM1657883 GSM1657884 GSM1657885 GSM1657886 GSM1657887 GSM1657888 GSM1657889 GSM1657890 GSM1657891 GSM1657892 GSM1657893 GSM1657894 GSM1657895 GSM1657896 GSM1657897 GSM1657898 GSM1657899 GSM1657900 GSM1657901 GSM1657902 GSM1657903 GSM1657904 GSM1657905 GSM1657906 GSM1657907 GSM1657908 GSM1657909 GSM1657910 GSM1657911 GSM1657912 GSM1657913 GSM1657914 GSM1657915 GSM1657916 GSM1657917 GSM1657918 GSM1657919 GSM1657920 GSM1657921 GSM1657922 GSM1657923 GSM1657924 GSM1657925 GSM1657926 GSM1657927 GSM1657928 GSM1657929 GSM1657930 GSM1657931 GSM1657932 GSM1657933 GSM1657934 GSM1657935 GSM1657936 GSM1657937 GSM1657938 GSM1657939 GSM1657940 GSM1657941 GSM1657942 GSM1657943 GSM1657944 GSM1657945 GSM1657946 GSM1657947 GSM1657948 GSM1657949 GSM1657950 GSM1657951 GSM1657952 GSM1657953 GSM1657954 GSM1657955 GSM1657956 GSM1657957 GSM1657958 GSM1657959 GSM1657960 GSM1657961 GSM1657962 GSM1657963 GSM1657964 GSM1657965 GSM1657966 GSM1657967 GSM1657968 GSM1657969 GSM1657970 GSM1657971 GSM1657972 GSM1657973 GSM1657974 GSM1657975 GSM1657976 GSM1657977 GSM1657978 GSM1657979 GSM1657980 GSM1657981 GSM1657982 GSM1657983 GSM1657984 GSM1657985 GSM1657986 GSM1657987 GSM1657988 GSM1657989 GSM1657990 GSM1657991 GSM1657992 GSM1657993 GSM1657994 GSM1657995 GSM1657996 GSM1657997 GSM1657998 GSM1657999 GSM1658000 GSM1658001 GSM1658002 GSM1658003 GSM1658004 GSM1658005 GSM1658006 GSM1658007 GSM1658008 GSM1658009 GSM1658010 GSM1658011 GSM1658012 GSM1658013 GSM1658014 GSM1658015 GSM1658016 GSM1658017 GSM1658018 GSM1658019 GSM1658020 GSM1658021 GSM1658022 GSM1658023 GSM1658024 GSM1658025 GSM1658026 GSM1658027 GSM1658028 GSM1658029 GSM1658030 GSM1658031 GSM1658032 GSM1658033 GSM1658034 GSM1658035 GSM1658036 GSM1658037 GSM1658038 GSM1658039 GSM1658040 GSM1658041 GSM1658042 GSM1658043 GSM1658044 GSM1658045 GSM1658046 GSM1658047 GSM1658048 GSM1658049 GSM1658050 GSM1658051 GSM1658052 GSM1658053 GSM1658054 GSM1658055 GSM1658056 GSM1658057 GSM1658058 GSM1658059 GSM1658060 GSM1658061 GSM1658062 GSM1658063 GSM1658064 GSM1658065 GSM1658066 GSM1658067 GSM1658068 GSM1658069 GSM1658070 GSM1658071 GSM1658072 GSM1658073 GSM1658074 GSM1658075 GSM1658076 GSM1658077 GSM1658078 GSM1658079 GSM1658080 GSM1658081 GSM1658082 GSM1658083 GSM1658084 GSM1658085 GSM1658086 GSM1658087 GSM1658088 GSM1658089 GSM1658090 GSM1658091 GSM1658092 GSM1658093 GSM1658094 GSM1658095 GSM1658096 GSM1658097 GSM1658098 GSM1658099 GSM1658100 GSM1658101 GSM1658102 GSM1658103 GSM1658104 GSM1658105 GSM1658106 GSM1658107 GSM1658108 GSM1658109 GSM1658110 GSM1658111 GSM1658112 GSM1658113 GSM1658114 GSM1658115 GSM1658116 GSM1658117 GSM1658118 GSM1658119 GSM1658120 GSM1658121 GSM1658122 GSM1658123 GSM1658124 GSM1658125 GSM1658126 GSM1658127 GSM1658128 GSM1658129 GSM1658130 GSM1658131 GSM1658132 GSM1658133 GSM1658134 GSM1658135 GSM1658136 GSM1658137 GSM1658138 GSM1658139 GSM1658140 GSM1658141 GSM1658142 GSM1658143 GSM1658144 GSM1658145 GSM1658146 GSM1658147 GSM1658148 GSM1658149 GSM1658150 GSM1658151 GSM1658152 GSM1658153 GSM1658154 GSM1658155 GSM1658156 GSM1658157 GSM1658158 GSM1658159 GSM1658160 GSM1658161 GSM1658162 GSM1658163 GSM1658164 GSM1658165 GSM1658166 GSM1658167 GSM1658168 GSM1658169 GSM1658170 GSM1658171 GSM1658172 GSM1658173 GSM1658174 GSM1658175 GSM1658176 GSM1658177 GSM1658178 GSM1658179 GSM1658180 GSM1658181 GSM1658182 GSM1658183 GSM1658184 GSM1658185 GSM1658186 GSM1658187 GSM1658188 GSM1658189 GSM1658190 GSM1658191 GSM1658192 GSM1658193 GSM1658194 GSM1658195 GSM1658196 GSM1658197 GSM1658198 GSM1658199 GSM1658200 GSM1658201 GSM1658202 GSM1658203 GSM1658204 GSM1658205 GSM1658206 GSM1658207 GSM1658208 GSM1658209 GSM1658210 GSM1658211 GSM1658212 GSM1658213 GSM1658214 GSM1658215 GSM1658216 GSM1658217 GSM1658218 GSM1658219 GSM1658220 GSM1658221 GSM1658222 GSM1658223 GSM1658224 GSM1658225 GSM1658226 GSM1658227 GSM1658228 GSM1658229 GSM1658230 GSM1658231 GSM1658232 GSM1658233 GSM1658234 GSM1658235 GSM1658236 GSM1658237 GSM1658238 GSM1658239 GSM1658240 GSM1658241 GSM1658242 GSM1658243 GSM1658244 GSM1658245 GSM1658246 GSM1658247 GSM1658248 GSM1658249 GSM1658251 GSM1658253 GSM1658255 GSM1658257 GSM1658259 GSM1658262 GSM1658264 GSM1658266 GSM1658268 GSM1658270 GSM1658272 GSM1658275 GSM1658277 GSM1658279 GSM1658281 GSM1658284 GSM1658286 GSM1658288 GSM1658290 GSM1658292 GSM1658294 GSM1658297 GSM1658299 GSM1658301 GSM1658304 GSM1658305 GSM1658306 GSM1658307 GSM1658308 GSM1658309 GSM1658310 GSM1658311 GSM1658312 GSM1658313 GSM1658314 GSM1658315 GSM1658316 GSM1658317 GSM1658318 GSM1658319 GSM1658320 GSM1658321 GSM1658322 GSM1658323 GSM1658324 GSM1658325 GSM1658326 GSM1658327 GSM1658328 GSM1658329 GSM1658330 GSM1658331 GSM1658332 GSM1658333 GSM1658334 GSM1658335 GSM1658336 GSM1658337 GSM1658338 GSM1658339 GSM1658340 GSM1658341 GSM1658342 GSM1658343 GSM1658344 GSM1658345 GSM1658346 GSM1658347 GSM1658348 GSM1658349 GSM1658350 GSM1658351 GSM1658352 GSM1658353 GSM1658354 GSM1658355 GSM1658356 GSM1658357 GSM1658358 GSM1658359 GSM1658360 GSM1658361 GSM1658362 GSM1658363 GSM1658364 GSM1658365 GSM1658366 \"\r\n",
      "!Series_contact_name\t\"Martin,,Enge\"\r\n"
     ]
    }
   ],
   "source": [
    "! head -n 20 /Users/kirkreardon/Downloads/*_series_matrix.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Whooo'"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"Whooo!!!!!!!!!\".strip(\"!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Whooo!!!!!!!!!'"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"Whooo!!!!!!!!!\".strip(\"o\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_title</th>\n",
       "      <th>Sample_geo_accession</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_submission_date</th>\n",
       "      <th>Sample_last_update_date</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Sample_channel_count</th>\n",
       "      <th>Sample_source_name_ch1</th>\n",
       "      <th>Sample_organism_ch1</th>\n",
       "      <th>Sample_characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>Sample_library_selection</th>\n",
       "      <th>Sample_library_source</th>\n",
       "      <th>Sample_library_strategy</th>\n",
       "      <th>Sample_relation</th>\n",
       "      <th>Sample_relation</th>\n",
       "      <th>Sample_supplementary_file_1</th>\n",
       "      <th>Sample_supplementary_file_2</th>\n",
       "      <th>series_matrix_table_begin</th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>series_matrix_table_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>healthy cortex cell 122</td>\n",
       "      <td>GSM1657992</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657992</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>healthy cortex cell 123</td>\n",
       "      <td>GSM1657993</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657993</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>healthy cortex cell 124</td>\n",
       "      <td>GSM1657994</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657994</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>healthy cortex cell 125</td>\n",
       "      <td>GSM1657995</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657995</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>healthy cortex cell 126</td>\n",
       "      <td>GSM1657996</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657996</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Sample_title Sample_geo_accession          Sample_status  \\\n",
       "1  healthy cortex cell 122           GSM1657992  Public on May 20 2015   \n",
       "2  healthy cortex cell 123           GSM1657993  Public on May 20 2015   \n",
       "3  healthy cortex cell 124           GSM1657994  Public on May 20 2015   \n",
       "4  healthy cortex cell 125           GSM1657995  Public on May 20 2015   \n",
       "5  healthy cortex cell 126           GSM1657996  Public on May 20 2015   \n",
       "\n",
       "  Sample_submission_date Sample_last_update_date Sample_type  \\\n",
       "1            Apr 15 2015             Nov 06 2015         SRA   \n",
       "2            Apr 15 2015             Nov 06 2015         SRA   \n",
       "3            Apr 15 2015             Nov 06 2015         SRA   \n",
       "4            Apr 15 2015             Nov 06 2015         SRA   \n",
       "5            Apr 15 2015             Nov 06 2015         SRA   \n",
       "\n",
       "  Sample_channel_count Sample_source_name_ch1 Sample_organism_ch1  \\\n",
       "1                    1                  Brain        Homo sapiens   \n",
       "2                    1                  Brain        Homo sapiens   \n",
       "3                    1                  Brain        Homo sapiens   \n",
       "4                    1                  Brain        Homo sapiens   \n",
       "5                    1                  Brain        Homo sapiens   \n",
       "\n",
       "  Sample_characteristics_ch1           ...           Sample_library_selection  \\\n",
       "1             tissue: cortex           ...                               cDNA   \n",
       "2             tissue: cortex           ...                               cDNA   \n",
       "3             tissue: cortex           ...                               cDNA   \n",
       "4             tissue: cortex           ...                               cDNA   \n",
       "5             tissue: cortex           ...                               cDNA   \n",
       "\n",
       "  Sample_library_source Sample_library_strategy  \\\n",
       "1        transcriptomic                 RNA-Seq   \n",
       "2        transcriptomic                 RNA-Seq   \n",
       "3        transcriptomic                 RNA-Seq   \n",
       "4        transcriptomic                 RNA-Seq   \n",
       "5        transcriptomic                 RNA-Seq   \n",
       "\n",
       "                                     Sample_relation  \\\n",
       "1  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "2  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "3  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "4  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "5  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "\n",
       "                                     Sample_relation  \\\n",
       "1  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "2  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "3  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "4  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "5  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "\n",
       "                         Sample_supplementary_file_1  \\\n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "5  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "\n",
       "                         Sample_supplementary_file_2  \\\n",
       "1  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "2  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "3  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "4  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "5  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "\n",
       "  series_matrix_table_begin      ID_REF series_matrix_table_end  \n",
       "1                       NaN  GSM1657992                     NaN  \n",
       "2                       NaN  GSM1657993                     NaN  \n",
       "3                       NaN  GSM1657994                     NaN  \n",
       "4                       NaN  GSM1657995                     NaN  \n",
       "5                       NaN  GSM1657996                     NaN  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata1 = pd.read_table('/Users/kirkreardon/Downloads/GSE67835-GPL15520_series_matrix.txt', \n",
    "                          skiprows=37, header=None, index_col=0)\n",
    "metadata1.index = metadata1.index.map(lambda x: x.strip('!'))\n",
    "# Transpose so each row is a cell\n",
    "metadata1 = metadata1.T\n",
    "metadata1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_title</th>\n",
       "      <th>Sample_geo_accession</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_submission_date</th>\n",
       "      <th>Sample_last_update_date</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Sample_channel_count</th>\n",
       "      <th>Sample_source_name_ch1</th>\n",
       "      <th>Sample_organism_ch1</th>\n",
       "      <th>Sample_characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>Sample_library_selection</th>\n",
       "      <th>Sample_library_source</th>\n",
       "      <th>Sample_library_strategy</th>\n",
       "      <th>Sample_relation</th>\n",
       "      <th>Sample_relation</th>\n",
       "      <th>Sample_supplementary_file_1</th>\n",
       "      <th>Sample_supplementary_file_2</th>\n",
       "      <th>series_matrix_table_begin</th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>series_matrix_table_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>healthy cortex cell 1</td>\n",
       "      <td>GSM1657871</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657871</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>healthy cortex cell 2</td>\n",
       "      <td>GSM1657872</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657872</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>healthy cortex cell 3</td>\n",
       "      <td>GSM1657873</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657873</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>healthy cortex cell 4</td>\n",
       "      <td>GSM1657874</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657874</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>healthy cortex cell 5</td>\n",
       "      <td>GSM1657875</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657875</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Sample_title Sample_geo_accession          Sample_status  \\\n",
       "1  healthy cortex cell 1           GSM1657871  Public on May 20 2015   \n",
       "2  healthy cortex cell 2           GSM1657872  Public on May 20 2015   \n",
       "3  healthy cortex cell 3           GSM1657873  Public on May 20 2015   \n",
       "4  healthy cortex cell 4           GSM1657874  Public on May 20 2015   \n",
       "5  healthy cortex cell 5           GSM1657875  Public on May 20 2015   \n",
       "\n",
       "  Sample_submission_date Sample_last_update_date Sample_type  \\\n",
       "1            Apr 15 2015             Nov 06 2015         SRA   \n",
       "2            Apr 15 2015             Nov 06 2015         SRA   \n",
       "3            Apr 15 2015             Nov 06 2015         SRA   \n",
       "4            Apr 15 2015             Nov 06 2015         SRA   \n",
       "5            Apr 15 2015             Nov 06 2015         SRA   \n",
       "\n",
       "  Sample_channel_count Sample_source_name_ch1 Sample_organism_ch1  \\\n",
       "1                    1                  Brain        Homo sapiens   \n",
       "2                    1                  Brain        Homo sapiens   \n",
       "3                    1                  Brain        Homo sapiens   \n",
       "4                    1                  Brain        Homo sapiens   \n",
       "5                    1                  Brain        Homo sapiens   \n",
       "\n",
       "  Sample_characteristics_ch1           ...           Sample_library_selection  \\\n",
       "1             tissue: cortex           ...                               cDNA   \n",
       "2             tissue: cortex           ...                               cDNA   \n",
       "3             tissue: cortex           ...                               cDNA   \n",
       "4             tissue: cortex           ...                               cDNA   \n",
       "5             tissue: cortex           ...                               cDNA   \n",
       "\n",
       "  Sample_library_source Sample_library_strategy  \\\n",
       "1        transcriptomic                 RNA-Seq   \n",
       "2        transcriptomic                 RNA-Seq   \n",
       "3        transcriptomic                 RNA-Seq   \n",
       "4        transcriptomic                 RNA-Seq   \n",
       "5        transcriptomic                 RNA-Seq   \n",
       "\n",
       "                                     Sample_relation  \\\n",
       "1  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "2  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "3  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "4  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "5  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "\n",
       "                                     Sample_relation  \\\n",
       "1  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "2  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "3  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "4  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "5  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "\n",
       "                         Sample_supplementary_file_1  \\\n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "5  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "\n",
       "                         Sample_supplementary_file_2  \\\n",
       "1  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "2  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "3  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "4  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "5  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "\n",
       "  series_matrix_table_begin      ID_REF series_matrix_table_end  \n",
       "1                       NaN  GSM1657871                     NaN  \n",
       "2                       NaN  GSM1657872                     NaN  \n",
       "3                       NaN  GSM1657873                     NaN  \n",
       "4                       NaN  GSM1657874                     NaN  \n",
       "5                       NaN  GSM1657875                     NaN  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata2 = pd.read_table('/Users/kirkreardon/Downloads/GSE67835-GPL18573_series_matrix.txt', \n",
    "                          skiprows=37, header=None, index_col=0)\n",
    "metadata2.index = metadata2.index.map(lambda x: x.strip('!'))\n",
    "# transpose\n",
    "metadata2 = metadata2.T\n",
    "metadata2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(466, 47)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_title</th>\n",
       "      <th>Sample_geo_accession</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_submission_date</th>\n",
       "      <th>Sample_last_update_date</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Sample_channel_count</th>\n",
       "      <th>Sample_source_name_ch1</th>\n",
       "      <th>Sample_organism_ch1</th>\n",
       "      <th>Sample_characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>Sample_library_selection</th>\n",
       "      <th>Sample_library_source</th>\n",
       "      <th>Sample_library_strategy</th>\n",
       "      <th>Sample_relation</th>\n",
       "      <th>Sample_relation</th>\n",
       "      <th>Sample_supplementary_file_1</th>\n",
       "      <th>Sample_supplementary_file_2</th>\n",
       "      <th>series_matrix_table_begin</th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>series_matrix_table_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>healthy cortex cell 122</td>\n",
       "      <td>GSM1657992</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657992</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>healthy cortex cell 123</td>\n",
       "      <td>GSM1657993</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657993</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>healthy cortex cell 124</td>\n",
       "      <td>GSM1657994</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657994</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>healthy cortex cell 125</td>\n",
       "      <td>GSM1657995</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657995</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>healthy cortex cell 126</td>\n",
       "      <td>GSM1657996</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657996</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Sample_title Sample_geo_accession          Sample_status  \\\n",
       "1  healthy cortex cell 122           GSM1657992  Public on May 20 2015   \n",
       "2  healthy cortex cell 123           GSM1657993  Public on May 20 2015   \n",
       "3  healthy cortex cell 124           GSM1657994  Public on May 20 2015   \n",
       "4  healthy cortex cell 125           GSM1657995  Public on May 20 2015   \n",
       "5  healthy cortex cell 126           GSM1657996  Public on May 20 2015   \n",
       "\n",
       "  Sample_submission_date Sample_last_update_date Sample_type  \\\n",
       "1            Apr 15 2015             Nov 06 2015         SRA   \n",
       "2            Apr 15 2015             Nov 06 2015         SRA   \n",
       "3            Apr 15 2015             Nov 06 2015         SRA   \n",
       "4            Apr 15 2015             Nov 06 2015         SRA   \n",
       "5            Apr 15 2015             Nov 06 2015         SRA   \n",
       "\n",
       "  Sample_channel_count Sample_source_name_ch1 Sample_organism_ch1  \\\n",
       "1                    1                  Brain        Homo sapiens   \n",
       "2                    1                  Brain        Homo sapiens   \n",
       "3                    1                  Brain        Homo sapiens   \n",
       "4                    1                  Brain        Homo sapiens   \n",
       "5                    1                  Brain        Homo sapiens   \n",
       "\n",
       "  Sample_characteristics_ch1           ...           Sample_library_selection  \\\n",
       "1             tissue: cortex           ...                               cDNA   \n",
       "2             tissue: cortex           ...                               cDNA   \n",
       "3             tissue: cortex           ...                               cDNA   \n",
       "4             tissue: cortex           ...                               cDNA   \n",
       "5             tissue: cortex           ...                               cDNA   \n",
       "\n",
       "  Sample_library_source Sample_library_strategy  \\\n",
       "1        transcriptomic                 RNA-Seq   \n",
       "2        transcriptomic                 RNA-Seq   \n",
       "3        transcriptomic                 RNA-Seq   \n",
       "4        transcriptomic                 RNA-Seq   \n",
       "5        transcriptomic                 RNA-Seq   \n",
       "\n",
       "                                     Sample_relation  \\\n",
       "1  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "2  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "3  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "4  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "5  BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "\n",
       "                                     Sample_relation  \\\n",
       "1  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "2  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "3  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "4  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "5  SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "\n",
       "                         Sample_supplementary_file_1  \\\n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "5  ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "\n",
       "                         Sample_supplementary_file_2  \\\n",
       "1  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "2  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "3  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "4  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "5  ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "\n",
       "  series_matrix_table_begin      ID_REF series_matrix_table_end  \n",
       "1                       NaN  GSM1657992                     NaN  \n",
       "2                       NaN  GSM1657993                     NaN  \n",
       "3                       NaN  GSM1657994                     NaN  \n",
       "4                       NaN  GSM1657995                     NaN  \n",
       "5                       NaN  GSM1657996                     NaN  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataframes = [metadata1, metadata2]\n",
    "metadata = pd.concat(dataframes)\n",
    "print(metadata.shape)\n",
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_title</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_submission_date</th>\n",
       "      <th>Sample_last_update_date</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Sample_channel_count</th>\n",
       "      <th>Sample_source_name_ch1</th>\n",
       "      <th>Sample_organism_ch1</th>\n",
       "      <th>Sample_characteristics_ch1</th>\n",
       "      <th>Sample_characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>Sample_library_selection</th>\n",
       "      <th>Sample_library_source</th>\n",
       "      <th>Sample_library_strategy</th>\n",
       "      <th>Sample_relation</th>\n",
       "      <th>Sample_relation</th>\n",
       "      <th>Sample_supplementary_file_1</th>\n",
       "      <th>Sample_supplementary_file_2</th>\n",
       "      <th>series_matrix_table_begin</th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>series_matrix_table_end</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_geo_accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM1657992</th>\n",
       "      <td>healthy cortex cell 122</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>cell type: astrocytes</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657992</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1657993</th>\n",
       "      <td>healthy cortex cell 123</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>cell type: endothelial</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657993</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1657994</th>\n",
       "      <td>healthy cortex cell 124</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>cell type: microglia</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657994</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1657995</th>\n",
       "      <td>healthy cortex cell 125</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>cell type: endothelial</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657995</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1657996</th>\n",
       "      <td>healthy cortex cell 126</td>\n",
       "      <td>Public on May 20 2015</td>\n",
       "      <td>Apr 15 2015</td>\n",
       "      <td>Nov 06 2015</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>tissue: cortex</td>\n",
       "      <td>cell type: microglia</td>\n",
       "      <td>...</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: http://www.ncbi.nlm.nih.gov/biosamp...</td>\n",
       "      <td>SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...</td>\n",
       "      <td>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM1657996</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Sample_title          Sample_status  \\\n",
       "Sample_geo_accession                                                   \n",
       "GSM1657992            healthy cortex cell 122  Public on May 20 2015   \n",
       "GSM1657993            healthy cortex cell 123  Public on May 20 2015   \n",
       "GSM1657994            healthy cortex cell 124  Public on May 20 2015   \n",
       "GSM1657995            healthy cortex cell 125  Public on May 20 2015   \n",
       "GSM1657996            healthy cortex cell 126  Public on May 20 2015   \n",
       "\n",
       "                     Sample_submission_date Sample_last_update_date  \\\n",
       "Sample_geo_accession                                                  \n",
       "GSM1657992                      Apr 15 2015             Nov 06 2015   \n",
       "GSM1657993                      Apr 15 2015             Nov 06 2015   \n",
       "GSM1657994                      Apr 15 2015             Nov 06 2015   \n",
       "GSM1657995                      Apr 15 2015             Nov 06 2015   \n",
       "GSM1657996                      Apr 15 2015             Nov 06 2015   \n",
       "\n",
       "                     Sample_type Sample_channel_count Sample_source_name_ch1  \\\n",
       "Sample_geo_accession                                                           \n",
       "GSM1657992                   SRA                    1                  Brain   \n",
       "GSM1657993                   SRA                    1                  Brain   \n",
       "GSM1657994                   SRA                    1                  Brain   \n",
       "GSM1657995                   SRA                    1                  Brain   \n",
       "GSM1657996                   SRA                    1                  Brain   \n",
       "\n",
       "                     Sample_organism_ch1 Sample_characteristics_ch1  \\\n",
       "Sample_geo_accession                                                  \n",
       "GSM1657992                  Homo sapiens             tissue: cortex   \n",
       "GSM1657993                  Homo sapiens             tissue: cortex   \n",
       "GSM1657994                  Homo sapiens             tissue: cortex   \n",
       "GSM1657995                  Homo sapiens             tissue: cortex   \n",
       "GSM1657996                  Homo sapiens             tissue: cortex   \n",
       "\n",
       "                     Sample_characteristics_ch1           ...            \\\n",
       "Sample_geo_accession                                      ...             \n",
       "GSM1657992                cell type: astrocytes           ...             \n",
       "GSM1657993               cell type: endothelial           ...             \n",
       "GSM1657994                 cell type: microglia           ...             \n",
       "GSM1657995               cell type: endothelial           ...             \n",
       "GSM1657996                 cell type: microglia           ...             \n",
       "\n",
       "                     Sample_library_selection Sample_library_source  \\\n",
       "Sample_geo_accession                                                  \n",
       "GSM1657992                               cDNA        transcriptomic   \n",
       "GSM1657993                               cDNA        transcriptomic   \n",
       "GSM1657994                               cDNA        transcriptomic   \n",
       "GSM1657995                               cDNA        transcriptomic   \n",
       "GSM1657996                               cDNA        transcriptomic   \n",
       "\n",
       "                     Sample_library_strategy  \\\n",
       "Sample_geo_accession                           \n",
       "GSM1657992                           RNA-Seq   \n",
       "GSM1657993                           RNA-Seq   \n",
       "GSM1657994                           RNA-Seq   \n",
       "GSM1657995                           RNA-Seq   \n",
       "GSM1657996                           RNA-Seq   \n",
       "\n",
       "                                                        Sample_relation  \\\n",
       "Sample_geo_accession                                                      \n",
       "GSM1657992            BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "GSM1657993            BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "GSM1657994            BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "GSM1657995            BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "GSM1657996            BioSample: http://www.ncbi.nlm.nih.gov/biosamp...   \n",
       "\n",
       "                                                        Sample_relation  \\\n",
       "Sample_geo_accession                                                      \n",
       "GSM1657992            SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "GSM1657993            SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "GSM1657994            SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "GSM1657995            SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "GSM1657996            SRA: http://www.ncbi.nlm.nih.gov/sra?term=SRX9...   \n",
       "\n",
       "                                            Sample_supplementary_file_1  \\\n",
       "Sample_geo_accession                                                      \n",
       "GSM1657992            ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "GSM1657993            ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "GSM1657994            ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "GSM1657995            ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "GSM1657996            ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supple...   \n",
       "\n",
       "                                            Sample_supplementary_file_2  \\\n",
       "Sample_geo_accession                                                      \n",
       "GSM1657992            ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "GSM1657993            ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "GSM1657994            ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "GSM1657995            ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "GSM1657996            ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-insta...   \n",
       "\n",
       "                     series_matrix_table_begin      ID_REF  \\\n",
       "Sample_geo_accession                                         \n",
       "GSM1657992                                 NaN  GSM1657992   \n",
       "GSM1657993                                 NaN  GSM1657993   \n",
       "GSM1657994                                 NaN  GSM1657994   \n",
       "GSM1657995                                 NaN  GSM1657995   \n",
       "GSM1657996                                 NaN  GSM1657996   \n",
       "\n",
       "                     series_matrix_table_end  \n",
       "Sample_geo_accession                          \n",
       "GSM1657992                               NaN  \n",
       "GSM1657993                               NaN  \n",
       "GSM1657994                               NaN  \n",
       "GSM1657995                               NaN  \n",
       "GSM1657996                               NaN  \n",
       "\n",
       "[5 rows x 46 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata = metadata.set_index('Sample_geo_accession')\n",
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir -p ~/projects/darmanis2015/processed_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "expression.to_csv('~/projects/darmanis2015/processed_data/expression.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "metadata.to_csv('~/projects/darmanis2015/processed_data/metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1/2-SBSRNA4                  0\n",
       "A1BG                         0\n",
       "A1BG-AS1                     0\n",
       "A1CF                         0\n",
       "A2LD1                        0\n",
       "A2M                         12\n",
       "A2ML1                        0\n",
       "A2MP1                        0\n",
       "A4GALT                       0\n",
       "A4GNT                        0\n",
       "AA06                         0\n",
       "AAA1                         0\n",
       "AAAS                         0\n",
       "AACS                         0\n",
       "AACSP1                       0\n",
       "AADAC                        0\n",
       "AADACL2                      0\n",
       "AADACL3                      0\n",
       "AADACL4                      0\n",
       "AADAT                        0\n",
       "AAGAB                        0\n",
       "AAK1                        36\n",
       "AAMP                         2\n",
       "AANAT                        0\n",
       "AARS                         4\n",
       "AARS2                        0\n",
       "AARSD1                       8\n",
       "AASDH                        4\n",
       "AASDHPPT                    29\n",
       "AASS                         0\n",
       "                         ...  \n",
       "ZSCAN22                      0\n",
       "ZSCAN23                      0\n",
       "ZSCAN29                     36\n",
       "ZSCAN30                      0\n",
       "ZSCAN4                       0\n",
       "ZSCAN5A                      0\n",
       "ZSCAN5B                      0\n",
       "ZSWIM1                       0\n",
       "ZSWIM2                       0\n",
       "ZSWIM3                       0\n",
       "ZSWIM4                       0\n",
       "ZSWIM5                       0\n",
       "ZSWIM6                       0\n",
       "ZSWIM7                      87\n",
       "ZUFSP                        2\n",
       "ZW10                         0\n",
       "ZWILCH                       0\n",
       "ZWINT                        0\n",
       "ZXDA                         0\n",
       "ZXDB                         0\n",
       "ZXDC                         2\n",
       "ZYG11A                       0\n",
       "ZYG11B                       1\n",
       "ZYX                          2\n",
       "ZZEF1                       84\n",
       "ZZZ3                         0\n",
       "tAKR                         0\n",
       "no_feature              824714\n",
       "ambiguous                 3241\n",
       "alignment_not_unique    596751\n",
       "Name: GSM1657884, dtype: int64"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression.GSM1657884"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['1/2-SBSRNA4', 'A1BG', 'A1BG-AS1', 'A1CF', 'A2LD1', 'A2M', 'A2ML1',\n",
       "       'A2MP1', 'A4GALT', 'A4GNT',\n",
       "       ...\n",
       "       'ZWINT', 'ZXDA', 'ZXDB', 'ZXDC', 'ZYG11A', 'ZYG11B', 'ZYX', 'ZZEF1',\n",
       "       'ZZZ3', 'tAKR'],\n",
       "      dtype='object', length=22085)"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bad_rows = ['no_feature', 'ambiguous', 'alignment_not_unique']\n",
    "good_genes = expression.index[~expression.index.isin(bad_rows)]\n",
    "good_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22088, 466)"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22085, 466)"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_actually_genes = expression.loc[good_genes]\n",
    "expression_actually_genes.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSM1657871</th>\n",
       "      <th>GSM1657872</th>\n",
       "      <th>GSM1657873</th>\n",
       "      <th>GSM1657874</th>\n",
       "      <th>GSM1657875</th>\n",
       "      <th>GSM1657876</th>\n",
       "      <th>GSM1657877</th>\n",
       "      <th>GSM1657878</th>\n",
       "      <th>GSM1657879</th>\n",
       "      <th>GSM1657880</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM1658357</th>\n",
       "      <th>GSM1658358</th>\n",
       "      <th>GSM1658359</th>\n",
       "      <th>GSM1658360</th>\n",
       "      <th>GSM1658361</th>\n",
       "      <th>GSM1658362</th>\n",
       "      <th>GSM1658363</th>\n",
       "      <th>GSM1658364</th>\n",
       "      <th>GSM1658365</th>\n",
       "      <th>GSM1658366</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>15</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>122</td>\n",
       "      <td>0</td>\n",
       "      <td>222</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>66</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tAKR</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no_feature</th>\n",
       "      <td>822062</td>\n",
       "      <td>970221</td>\n",
       "      <td>952904</td>\n",
       "      <td>1301729</td>\n",
       "      <td>882830</td>\n",
       "      <td>1098392</td>\n",
       "      <td>768831</td>\n",
       "      <td>1436879</td>\n",
       "      <td>936055</td>\n",
       "      <td>729299</td>\n",
       "      <td>...</td>\n",
       "      <td>1136953</td>\n",
       "      <td>862137</td>\n",
       "      <td>402523</td>\n",
       "      <td>913528</td>\n",
       "      <td>592939</td>\n",
       "      <td>505954</td>\n",
       "      <td>750682</td>\n",
       "      <td>606953</td>\n",
       "      <td>1063098</td>\n",
       "      <td>791301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ambiguous</th>\n",
       "      <td>5663</td>\n",
       "      <td>5771</td>\n",
       "      <td>7831</td>\n",
       "      <td>4780</td>\n",
       "      <td>5002</td>\n",
       "      <td>2388</td>\n",
       "      <td>10386</td>\n",
       "      <td>8422</td>\n",
       "      <td>4667</td>\n",
       "      <td>11112</td>\n",
       "      <td>...</td>\n",
       "      <td>7730</td>\n",
       "      <td>4973</td>\n",
       "      <td>4168</td>\n",
       "      <td>5718</td>\n",
       "      <td>5249</td>\n",
       "      <td>5350</td>\n",
       "      <td>7668</td>\n",
       "      <td>7337</td>\n",
       "      <td>8792</td>\n",
       "      <td>4359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>alignment_not_unique</th>\n",
       "      <td>697959</td>\n",
       "      <td>847047</td>\n",
       "      <td>613329</td>\n",
       "      <td>742635</td>\n",
       "      <td>765874</td>\n",
       "      <td>1118933</td>\n",
       "      <td>556215</td>\n",
       "      <td>1051949</td>\n",
       "      <td>669653</td>\n",
       "      <td>584841</td>\n",
       "      <td>...</td>\n",
       "      <td>601759</td>\n",
       "      <td>525037</td>\n",
       "      <td>456087</td>\n",
       "      <td>570485</td>\n",
       "      <td>466276</td>\n",
       "      <td>465384</td>\n",
       "      <td>655463</td>\n",
       "      <td>460247</td>\n",
       "      <td>830121</td>\n",
       "      <td>730874</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  466 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       GSM1657871  GSM1657872  GSM1657873  GSM1657874  \\\n",
       "0                                                                       \n",
       "ZZZ3                           15          58           0         122   \n",
       "tAKR                            0           0           0           0   \n",
       "no_feature                 822062      970221      952904     1301729   \n",
       "ambiguous                    5663        5771        7831        4780   \n",
       "alignment_not_unique       697959      847047      613329      742635   \n",
       "\n",
       "                       GSM1657875  GSM1657876  GSM1657877  GSM1657878  \\\n",
       "0                                                                       \n",
       "ZZZ3                            0         222           2           0   \n",
       "tAKR                            0           0           0           0   \n",
       "no_feature                 882830     1098392      768831     1436879   \n",
       "ambiguous                    5002        2388       10386        8422   \n",
       "alignment_not_unique       765874     1118933      556215     1051949   \n",
       "\n",
       "                       GSM1657879  GSM1657880     ...      GSM1658357  \\\n",
       "0                                                 ...                   \n",
       "ZZZ3                           13           0     ...               0   \n",
       "tAKR                            0           0     ...               0   \n",
       "no_feature                 936055      729299     ...         1136953   \n",
       "ambiguous                    4667       11112     ...            7730   \n",
       "alignment_not_unique       669653      584841     ...          601759   \n",
       "\n",
       "                       GSM1658358  GSM1658359  GSM1658360  GSM1658361  \\\n",
       "0                                                                       \n",
       "ZZZ3                           66           0           0           0   \n",
       "tAKR                            0           0           0           0   \n",
       "no_feature                 862137      402523      913528      592939   \n",
       "ambiguous                    4973        4168        5718        5249   \n",
       "alignment_not_unique       525037      456087      570485      466276   \n",
       "\n",
       "                       GSM1658362  GSM1658363  GSM1658364  GSM1658365  \\\n",
       "0                                                                       \n",
       "ZZZ3                            0           4           0           0   \n",
       "tAKR                            0           0           0           0   \n",
       "no_feature                 505954      750682      606953     1063098   \n",
       "ambiguous                    5350        7668        7337        8792   \n",
       "alignment_not_unique       465384      655463      460247      830121   \n",
       "\n",
       "                       GSM1658366  \n",
       "0                                  \n",
       "ZZZ3                            7  \n",
       "tAKR                            0  \n",
       "no_feature                 791301  \n",
       "ambiguous                    4359  \n",
       "alignment_not_unique       730874  \n",
       "\n",
       "[5 rows x 466 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_actually_genes.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "expression_actually_genes.to_csv(\"/Users/kirkreardon/projects/darmanis2015/processed_data/expression_actually_genes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GSM1657871    int64\n",
       "GSM1657872    int64\n",
       "GSM1657873    int64\n",
       "GSM1657874    int64\n",
       "GSM1657875    int64\n",
       "GSM1657876    int64\n",
       "GSM1657877    int64\n",
       "GSM1657878    int64\n",
       "GSM1657879    int64\n",
       "GSM1657880    int64\n",
       "GSM1657881    int64\n",
       "GSM1657882    int64\n",
       "GSM1657883    int64\n",
       "GSM1657884    int64\n",
       "GSM1657885    int64\n",
       "GSM1657886    int64\n",
       "GSM1657887    int64\n",
       "GSM1657888    int64\n",
       "GSM1657889    int64\n",
       "GSM1657890    int64\n",
       "GSM1657891    int64\n",
       "GSM1657892    int64\n",
       "GSM1657893    int64\n",
       "GSM1657894    int64\n",
       "GSM1657895    int64\n",
       "GSM1657896    int64\n",
       "GSM1657897    int64\n",
       "GSM1657898    int64\n",
       "GSM1657899    int64\n",
       "GSM1657900    int64\n",
       "              ...  \n",
       "GSM1658337    int64\n",
       "GSM1658338    int64\n",
       "GSM1658339    int64\n",
       "GSM1658340    int64\n",
       "GSM1658341    int64\n",
       "GSM1658342    int64\n",
       "GSM1658343    int64\n",
       "GSM1658344    int64\n",
       "GSM1658345    int64\n",
       "GSM1658346    int64\n",
       "GSM1658347    int64\n",
       "GSM1658348    int64\n",
       "GSM1658349    int64\n",
       "GSM1658350    int64\n",
       "GSM1658351    int64\n",
       "GSM1658352    int64\n",
       "GSM1658353    int64\n",
       "GSM1658354    int64\n",
       "GSM1658355    int64\n",
       "GSM1658356    int64\n",
       "GSM1658357    int64\n",
       "GSM1658358    int64\n",
       "GSM1658359    int64\n",
       "GSM1658360    int64\n",
       "GSM1658361    int64\n",
       "GSM1658362    int64\n",
       "GSM1658363    int64\n",
       "GSM1658364    int64\n",
       "GSM1658365    int64\n",
       "GSM1658366    int64\n",
       "dtype: object"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_actually_genes.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ZZZ3 ', 'tAKR ', 'no_feature ', 'ambiguous ', 'alignment_not_unique '], dtype='object', name=0)"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_actually_genes.tail().index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (single-cell-bioinformatics)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
